

## Protocol

### General Procedures

**Vivarium**
* Our room in the mouse facility is room 1102 of the ICRC
* There are two hoods for use
* These hoods should be reserved in advance via [iLab](https://cumc.corefacilities.org)

**Sterility**

### 2021-03-18
**Mice**
* Miced arrived yesterday, 2021-03-17
* Giving 48hrs to acclimate to facility
* Will start with normalization process tmrw, 2021-03-19
* Placed service request for PI cage change; approved

**Preparing Metadata**
* initial [metadata csv](./metadata/samples.xlsx)
* Columns:
    * sample: The sample id, used to label the tubes and for the sequencing library
    * date: date the sample was collected
    * day : day of the sample protocol, where day 0 is when the mice are co-housed with their paired mouse
    * cage_num : the number of the cage, used to label the cage cards and to differentiate repeat condition cages
    * tx_group : The treatment group of each cage, either jax control, envigo control, or Jax X Envigo pairs
    * mouse_type : The facility of origin for each mouse
    * mouse_num : An identifier number for each mouse in the cage
    * mark : How the mice are physically marked to distinguish them. 
        * In each cage, mouse 1 has no tail markings and mouse 2 has 3 black stripes on their tail.
* Note: need to update with actual start date etc.


### 2021-03-19
**Initiating PI cage change (TM)**
* Add blue special request cards to each cage indicating PI-only cage change; start today
* Cards have JAX or Envigo on them
* Experiment listed as "GU FMT"

**Normalization (TM)**
* Started normalization process
* We have 2 cages w/ 4 mice from Jax; 2 cages w/4 mice from envigo
* Starting with Jax:
    * Thoroughly wiped down hood, allowed to sit wet for ~3min
    * Pulled both cages
    * Working w/ one cage open at a time, scopped ~50% of the bedding from a cage into a container; did for both cages
    * Mixed the bedding from both cages together thoroughly (~30sec)
    * Distributed the bedding equally between both cages
* Envigo following the same procedure w/ a thorough cleaning in-between & glove change

### 2021-03-22
**Normalization Round 2 (TM)**
* Started normalization process
* We have 2 cages w/ 4 mice from Jax; 2 cages w/4 mice from envigo
* Starting with Jax:
    * Thoroughly wiped down hood, allowed to sit wet for ~3min
    * Pulled both cages
    * Working w/ one cage open at a time, scopped ~50% of the bedding from a cage into a container; did for both cages
    * Mixed the bedding from both cages together thoroughly (~30sec)
    * Distributed the bedding equally between both cages
* Envigo following the same procedure w/ a thorough cleaning in-between & glove change

### 2021-03-24
**Normalization Round 3 Planning**
* For ease of future collection, we're planning to start our experiment on Monday, 2021-03-29
* This will give us an extra round of normalization, which we'll start tomorroe 2021-03-25
    * Round 1, 3 days
    * Round 2, 3 days
    * Round 3, 3 days
    * Initiate experiment 2021-03-29
* Collection plan
    * collect pellets into weighed tubes on ice
    * weigh post pellet to get pellet weight
    * Break up/vortex sample in PBS
    * Allow to settle on ice
    * Take 100uL of supernatant, aliquot for 16Sv4
    * Take 600uL of supernatant, move to new tube
        * Add 600uL of 40% glycerol
        * Aliquot 4X 200uL into PCR strips

**Preparing Documents**
* Had final convo w/ GU about plan
    * Final timeline - [timeline.csv](./metadata/timeline.csv) 
    * Final Sample table - [samples.csv](./metadata/samples.csv)
* Used a jupyter notebook to prepare printer logs and the overall sample table (from above)
    * jupyter - [0_create_print_logs.ipynb](./jupyter/0_create_print_logs.ipynb)
        * some pseudo-metadata is stored in  - [sample_collection](./documents/sample_collection/)
    * Print logs located by day at - [print_logs](./documents/sample_collection/print_logs)

**Preparing for Collection**
* printed log for d-3
* labeled and weight tubes for d0
* Filled out [samples.csv](./metadata/samples.csv) w/ tube weights

### 2021-03-25

**Normalization round 3 (GU)**
* Cages were formally labeled Jax 1/2; envigo 1;2
* 50% of Jax cage 1 bedding was transferred to Jax cage 2 and vice versa
* 50% of Envigo cage 1 bedding was transferred to Envigo cage 2 and vice versa

**Day -3 Collection (GU)**
* GU collected samples
* Mice were assigned mouse numbers and given tail markings
    * Note: Green marker was not working, switched to double blue dashes as a temp replacement 

**Sample Processing (GU/TM)**
* Sample were brough back on ice and weighed
* Moved into anaerobic chamber
* Put through processing:
    * add 1mL reduced sterile PBS
    * Mash pellet with pipette to break up; vortex
    * Allow to settle for ~3min on ice
    * draw off 100uL of supernatant -> store on ice; used for 16Sv4
    * draw off 600uL into unlabeled temp tube
    * add 600ul 40% glycerol to unlabeled temp tube; vortex
    * Aliquot 4X 200uL into labeled PCR strip tubes
* 16Sv4 samples & 200uL aliquots frozen at -80C

### 2021-03-26
#### Day -2 collection (GU)
* See GU notes

### 2021-03-28
#### day 0 collection (TM/GU)
* Collected and moved mice into their "FMT" cage for Jax vs. Envigo engraftment
* All collections onto ice for post-processing w. live bacteria
* General process:
    * Prepared 8 new cages and appropriate labels for them
    * In different hoods, normalization cage 1 (w/ 4 mice ea) for Jax and Envigo were collected
    * Then they were moved into their new experimental treatment cage; in this case all were same-vendor control pairs
    * In different hoods, norm. cage #2 for Jax/Envigo were collected for all 4 mice
    * The envigo mice were then distributed into 4 new cages; Jax X Envigo 1-4 (JE1-4)
    * These cages were then shuttled to the second hood where the remaining jax cage (Jax 2 - norm. phase cage) was and the matching jax mouse was added to the cage w. the envigo mouse
    * The now complete experimental tx group cage was added to the rack before proceeding with the next pairing

#### Sample processing (GU/TM)
* followed procedure from d-3
* GU worked downstairs on first 8 sample; TM worked on 3rd floor w/ second 8 samples
* Stored all samples at -80C

#### Updating Logs
* Updated sample log to include weights for d-2 and d0
* note that the tubes for 37 and 41 were mixed up & re-labeled. Weights are appropriately switched in the sample log
* Uploaded scans for [d-2](./documents/scans/day_-2_log.pdf) and [d0](./documents/scans/day_0_log.pdf) sample weight logs


### 2021-03-29
#### Day 1 preparation
* Labeled and weighed next set of pellet sample tubes - see [print sheet](./documents/sample_collection/print_logs/day1.csv), printed two copies
* Labeled a set of glycerol/transfer & final 16Sv4 sample tubes, split for GU/TM concurrent processing 
* Labeled a set of PCR tubes, split for GU/TM concurrent processing 

#### Day 1 Collection
* Normal collection
* GU -> Jax & Env. Controls; TM -> Jax X Env.

#### Sample Processing
* Normal procedure
* GU/TM processed samples they collected;
* weights updated in samples sheet
* scannned log added uploaded

### 2021-03-30
#### Preparation (GU)
* Prepared all tubes for collection / processing

#### d2 Sample Collection (GU/TUM)
* Collected w/ normal protocol
* GU jax & envigo controls; TUM Jax Envigo cohouse sample

#### Sample Processing
* GU, TUM processed samples they collected
* Also stored 16 samples of the excess 20% glycerol mix generated at -80C in a new box
* Uploaded scanned sample weight sheet
* updates samples.csv metadata w/ today's weights

### 2021-03-31
#### Day 3 Preparation
* Preparing 16 weighted labeled tubes for pellet collection
* Also labeling 2X 16 tubes for glycerol mixing & 16S storage

#### Day 3 Sample Collection (TM)
* collected samples per usual protocol

### 2021-04-01
#### Sample collection (TM)
* Samples collected per usual protocol
* time ~45mins!
* weighed samples -> updated samples.csv, added uploaded scanned document
* moved to 4C in ice bucket inside sample blocks; GU will pickup for processing

#### Adding data for d3 collection
* weights were calc'd yesterday, didn't add to samples.csv or upload scan
* Did both

#### Paperwork stuff
* Need to figure out cage change date
* start 2021-03-28 -> cage changed here 
* do next cage change monday, 2021-04-05; re-up markings
* Update cage cards -> Add that we cage changed monday, include start date of 2021-03-19

#### Preparing d5 collection
* label and weight out 16 pellet tubes
* prepare 2x 16 labeled tubes for glycerol; seq stock
* print d5 log to record

### 2021-04-02
#### D5 Collection (TM)
* Collected pellets, normal protocol
* Re-did all tail markings for mice; planning to cage change Monday and will re-up again then.
* weighed tubes and handed off to GU for processing
* entered sample weights into samples.csv; uplaoded scanned weight log


### 2021-04-13
#### D15 Collection
* Labeled and weighed 3 sets of sterile 1.5mL tubes; weighed 16S set
* Collected pellets onto ice from mice following normal sterile procedure
* Weighed pellets; updated `samples.csv` & uploaded scanned weight document for today

#### Preparing for D16/17 Collection
* Prepared labeled 3X 1.5mL for both day 16/17, weighed one set for 16S sample storage
* Prepared 50mL stock of 40% glycerol & Sterile PBS; move to chamber to reduce

### 2021-04-14
#### D16 collecton
* Collected pellets onto ice from mice following normal sterile procedure
* Weighed pellets; updated `samples.csv` & uploaded scanned weight document for today
* Note: noticed a difference in pellets between groups:
> We've talked about how the the pellets between jax and envigo are crazy different. 
Envigo is much softer, lighter in color, and I guestimate generally larger vs. the dryer, 
"lumpy" pellets from Jax. Is there any kinda ez analytical methods we could use to 
figure out what's diff. between the two?

### 2021-04-15
#### D17 Collection
* Collected pellets onto ice from mice following normal sterile procedure
* Weighed pellets; updated `samples.csv` & uploaded scanned weight document for today

### 2021-04-26
#### Final Collection Schedule
* will collect 
    * d30 - 2021-04-28
    * d31 - 2021-04-29
    * d32 - 2021-04-30, sac day w/ tissue collection
* Updated `samples.csv` by hand w/ d30; d31
* updated `print_logs` for d30;d31 for this too
* Created `sac_samples.csv` with details for samples collected on sac day
    * starting w/ jxe.209 as first number
    * For each mouse:
        * collecting Terminal Ileum, Cecum, and transverse Colon samples
        * 16S - Sampling luminal content, mucus layer via scraping
        * RNAseq - Intestinal tissue


### 2021-04-28
#### d30 collection
* Collected samples, weighed them out. updated `samples.csv` and uploaded scan log
* Noted 2021-04-26 that several mice had bald spots while re-upping tail markings
* recorded today
    * EC1 no tail marking - bald spot on back
    * JE1 no tail marking - bald spot of back
    * JE2 black tail marking - bald snout, whiskers fell out

### 2021-04-29
#### d31 collection
* collected samples, weighed them out. Updates `samples.csv`, uploaded scan log


#### Preparing tubes for D32 Collection + Sac'ing
* Prepared a seperate sample log for the samples collected post-sac
    * A base sample info sheet to build off of : [sac_samples_base.csv](./documents/sac_sample_collection/sac_samples_base.csv)
    * A matrix of the type of samples to be collected for each mouse [collection_map.csv](./documents/sac_sample_collection//collection_map.csv)
    * Jupyternotebook for putting together final sample log [1_create_sac_samples_list.ipynb](./jupyter/1_create_sac_samples_list.ipynb)
    * sample log : [sac_samples.csv](./metadata/sac_samples.csv)

* Prepared & labeled autoclaved tubes
* Set aside MaPS-seq2 (dots) and tissues RNAseq tubes (dots)
* Weighed remaining tubes for 16S/Shotgun Metagenomics
* entered into `sac_samples.csv`

#### Preparing MaPS-seq2 tubes
* Made methacarn (60% methanol, 30% chloroform, 10% acetic acid) for 24 samples (1mL ea, 30mL total w/ excess)
    * 18mL Methanol
    * 9mL Chloroform
    * 3mL Acetic Acid
* Aliquoted 1mL into each labeled tube

#### Preparing Tissue RNAseq tubes
* Borrowing RNAlater from CM
* Laminar flow hood, EtOH then RNAzap sterilized
* Aliquoting 500uL into each labeled tube


### 2021-05-10
#### TM 16S Processing Pipeline
* raw reads located on S3 at [JxE_FMT_Redo-257630374](s3://guillaume-urtecho/20210510_FMT_redo_16S/raw_data/JxE_FMT_Redo-257630374/)
* TM analysis on S3 at [2021-03-18_JxE_replication](s3://thomas-moody/fmt/2021-03-18_JxE_replication/)
* Loading AWS w/ tum AMI, following on AWS
* make and use project folder : `mkdir jxe; cd jxe`
* make a seq + raw_reads folder : `mkdir -p seq/00_raw_reads`
* Copy raw reads to seq folder
* `aws s3 cp s3://guillaume-urtecho/20210510_FMT_redo_16S/raw_data/JxE_FMT_Redo-257630374 seq/00_raw_reads --recursive`
* Rename reads to match metadata
* Script : [rename_seq_files.py](s3://thomas-moody/fmt/2021-03-18_JxE_replication/scripts/rename_seq_files.py)
* Processing completed w/ Snakemake
* Copying multiqc report to local
* `scp -i ~/Dropbox/aws_pems/tum.pem ec2-user@52.72.35.224:/home/ec2-user/jxe/seq/02_qc/multiqc_report.html seq/02_qc`
* Looks... ok?
* Copying OTU table and TAX table processing scripts
    * `aws s3 cp s3://thomas-moody/cibio/2021-03-08_cq_biogeography/scripts/process_otu.py scripts`
    * `aws s3 cp s3://thomas-moody/cibio/2021-03-08_cq_biogeography/scripts/process_tax.py scripts`
* Copying existing sintax db
    * `mkdir db`
    * `aws s3 cp s3://thomas-moody/cibio/2021-03-08_cq_biogeography/db/sintax.udb db`
* Complete!
* Upload complete pipeline and set of files produced to s3 folder
* `aws s3 sync . s3://thomas-moody/fmt/2021-03-18_JxE_replication/`

#### TM Local Analysis & Viz
* Pulling down important files
    * `mkdir data`
    * 
    * `scp -i ~/Dropbox/aws_pems/tum.pem ec2-user@52.72.35.224:/home/ec2-user/jxe/seq/09_table/zotu.csv data`
    * `scp -i ~/Dropbox/aws_pems/tum.pem ec2-user@52.72.35.224:/home/ec2-user/jxe/seq/09_table/tax.csv data`
    * `scp -i ~/Dropbox/aws_pems/tum.pem ec2-user@52.72.35.224:/home/ec2-user/jxe/seq/07_zotu/zotus.fa ./data`

### 2021-05-30
#### Re-Running 16S Pipeline Across all FMT Experiments
* Added old FMT experiments to shared directory on EC2
* Going to upload the necc. files for this experiment as well
* making experiment dir & copying over 16S data from pellets
```
mkdir 2021-03-18_JxE_FMT03
cd 2021-03-18_JxE_FMT03
mkdir -p seq/00_raw_reads
mkdir metadata
mv ~/jxe/seq/00_raw_reads/jxe.*.fastq.gz seq/00_raw_reads/
```
* Copying reads from recent 16S run for the luminal content samples
* uploading metadata sheet with s3 paths for GU bucket to EC2 -> [seq_sac_samples.csv](metadata/seq_sac_samples.csv)
    * `scp -i ~/Dropbox/aws_pems/tum.pem metadata/seq_sac_samples.csv ec2-user@52.72.35.224:~/fmt/experiments/2021-03-18_JxE_FMT03/metadata`
* created python scripts on EC2 `scripts/00_download_seq_files.py`
