---
title: "Fig1_FMT"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

setwd("~/Dropbox/fmt/scripts/")
library(tidyverse)
library(wesanderson)
library(ggthemes)
library(cowplot)
#library(svglite)
library('phyloseq')

ven <- c('#FAAF40',"#1B75BB", "#37B34A","#F05A28")
pal <- wes_palette("Zissou1", 100, type = "continuous")

options(stringsAsFactors = F)
options(scipen = 10000)

```

Load data

```{r cars}

zotu <- read_csv("../seq/zotu.csv") 
tax <- read_csv("../seq/tax.csv")
meta <- read_csv("../experiments/2017-10-10_pairwise_cohousing/metadata/samples.csv")

# Check read counts
read.counts <- zotu %>%
  rowwise(seqid) %>%
  mutate(numreads = rowSums(across(where(is.numeric)))) %>%
  ungroup() %>%
  select(seqid, numreads)

# Pull out samples w/ >10k reads
samples.to.keep <- read.counts %>%
  filter(numreads > 1000) %>%
  select(seqid) %>%
  as_vector()

# Drop <10k read samples from zotu table
zotu <- zotu %>%
  filter(seqid %in% samples.to.keep)

# make tidy data
# Convert to relative abundance
zotu <- zotu %>%
  pivot_longer(!seqid, names_to="OTU", values_to="reads") %>%
  group_by(seqid) %>%
  mutate(relabd = (reads / sum(reads))) %>% ungroup
#  mutate(relabd = (reads / sum(reads))*1000) %>% ungroup


# calc log10 OTU abd
zotu <- zotu %>%
  mutate(log10relabd = if_else(relabd == 0, -6, log10(relabd)))



# Join metadata to zotu table
zotu <- zotu %>%
  left_join(meta, by="seqid")

gtable <- tax %>%
  select(OTU, genus, family)
```

filter out for only OTUs with sufficient abundance in any mouse type

```{r}

jax.otus.to.keep <- zotu %>%
  filter(mouse_type == "Jackson") %>%
  group_by(OTU) %>%
  summarise(avg_abd = mean(relabd)) %>%
  filter(avg_abd > .005) %>%
  select(OTU) %>%
  as_vector()

envigo.otus.to.keep <- zotu %>%
  filter(mouse_type == "Envigo") %>%
  group_by(OTU) %>%
  summarise(avg_abd = mean(relabd)) %>%
  filter(avg_abd > .005) %>%
  select(OTU) %>%
  as_vector()

charles.otus.to.keep <- zotu %>%
  filter(mouse_type == "Charles River") %>%
  group_by(OTU) %>%
  summarise(avg_abd = mean(relabd)) %>%
  filter(avg_abd > .005) %>%
  select(OTU) %>%
  as_vector()

taconic.otus.to.keep <- zotu %>%
  filter(mouse_type == "Taconic") %>%
  group_by(OTU) %>%
  summarise(avg_abd = mean(relabd)) %>%
  filter(avg_abd > .005) %>%
  select(OTU) %>%
  as_vector()

otus.to.keep <- unique(c(jax.otus.to.keep, envigo.otus.to.keep,
                         charles.otus.to.keep, taconic.otus.to.keep))

# Drop OTUs not at least 1% abd
zotu <- zotu %>%
  filter(OTU %in% otus.to.keep)


```


Fig 1A - Showing different vendor

```{r}

library('ggdendro')
library('grid')

set.seed(123)

vendor <- zotu %>% filter(day == 0) %>%
  select(OTU, mouse, mouse_type, log10relabd) %>% reshape2::dcast(OTU~mouse)



vendor.scaled <- vendor
#vendor.scaled[, c(2:41)] <- (vendor.scaled[, 2:41]) #don't scale here actually

#clustering
vendor.matrix <- as.matrix(vendor.scaled[, -c(1)])
rownames(vendor.matrix) <- vendor.scaled$OTU
vendor.dendro <- as.dendrogram(hclust(d = dist(x = vendor.matrix)))

# Create dendro
dendro.plot <- ggdendrogram(data = vendor.dendro, rotate = TRUE)

# Preview the plot
print(dendro.plot)


##Heatmap
vendor.long <- reshape2::melt(vendor.scaled, id = c("OTU"))
vendor.order <- order.dendrogram(vendor.dendro) #reorder
vendor.long <- left_join(vendor.long, gtable) %>% #add family data
    left_join(., (select(zotu, mouse, mouse_type) %>% distinct()), by = c('variable'='mouse')) #add vendor data
vendor.long$OTU <- factor(x = vendor.long$OTU,
                               levels = vendor.scaled$OTU[vendor.order], 
                               ordered = TRUE)

#order vendors for plot
vendor.long$mouse_type <- factor(vendor.long$mouse_type,      # Reordering group factor levels
                         levels = c("Taconic","Jackson", "Charles River", "Envigo"))
# Heatmap
heatmap.plot <- ggplot(data = vendor.long, aes(x = variable, y = OTU)) +
  geom_tile(aes(fill = value), color = 'white') +
  scale_fill_gradient2(high = 'firebrick3', mid = 'white', low = '#1357a6', midpoint = -3) +
  #scale_fill_gradientn(colours = pal) + 
  labs(x='Mouse (N = 10 / Vendor)', fill ='Relative Abundance',
       y = paste("OTU (N = ", length(otus.to.keep), ")", sep = '')) +
  facet_wrap(~mouse_type, scales = 'free_x', ncol = 4) +
  theme(strip.text.y.right = element_text(angle = 0),
        panel.background = element_rect(fill = "white"), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(), 
              axis.text.x = element_blank(),
              strip.text = element_text(size = 10),
              panel.spacing = unit(0.05, "lines"),
        legend.position = 'bottom',
        strip.background = element_blank()) 

fam <- ggplot(data = vendor.long, aes(x = '', y = OTU)) + 
    geom_tile(aes(fill = family), color = 'white') +
    labs(fill = 'Family') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_rect(fill = "white"), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))

heatmap.plot
#cowplot::plot_grid(heatmap.plot, fam, ncol =2, rel_widths = c(.6,.2))
ggsave("../figs/fig1/Vendor16S.png",  width = 5, height = 5)
ggsave("../figs/fig1/Vendor16S.svg",  width = 5, height = 5)

```


Fig 1B - PCA of mouse vendor from 16S data
```{r , echo=FALSE}

library(reshape2)
library(M3C)

vendor <- zotu %>% filter(day == 0, OTU != 'Otu10') %>% #Look @ D0 samples, remove sporo spike-in
  select(OTU, seqid,mouse_type, relabd) %>% #mutate(relabd=log10(relabd)) %>%
  reshape2::dcast(., mouse_type + seqid ~ OTU) 

grp_key <- vendor[,1:2]

#create such that samples as columns, rows as features
tSNE_header <- (vendor %>% select(-seqid) %>% t() %>% as.data.frame())[1,]


tSNE_input <- (vendor %>% select(-seqid) %>% t() %>% as.data.frame())[-1,] %>% 
              mutate_all(., function(x) as.numeric(as.character(x))) 

names(tSNE_input) <- tSNE_header[1,]


M3C::pca(tSNE_input, 
         labels = as.factor(names(tSNE_input)),
         colvec = ven,dotsize = 3,
         controlscale = T, scale = 3, 
         axistextsize = 10,
         legendtextsize = 14, legendtitle = '')

ggsave("../figs/fig1/1B_VendorPCA.png", width = 4.5, height = 2.5)
ggsave("../figs/fig1/1B_VendorPCA.pdf", width = 4.5, height = 2.5)

        
```

Same with unifrac
```{r}

library(msa)
library(ape)

vendor <- zotu %>% filter(day == 0, OTU != 'Otu10') %>% #Look @ D0 samples, remove sporo spike-in
  select(OTU, sample, reads) %>% #mutate(relabd=log10(relabd)) %>%
  reshape2::dcast(., sample ~ OTU) 

#grp_key <- vendor[,1:2]

#create such that samples as columns, rows as features
tSNE_header <- (vendor %>% t() %>% as.data.frame())[1,]


tSNE_input <- (vendor %>% t() %>% as.data.frame())[-1,] %>% 
              mutate_all(., function(x) as.numeric(as.character(x))) 

names(tSNE_input) <- tSNE_header[1,]


#now make taxa table
taxmat <- tax %>% select(OTU, phylum, class, order, family, genus) %>% 
  column_to_rownames('OTU') %>% as.matrix()

#get unifrac distances
#Generate MSA
MUSCLE_out <- readDNAStringSet('../seq/cleaned_zotus.fa') %>% msaMuscle(.)

#Convert alignment for seqinr package
myAln2 <- msaConvert(MUSCLE_out, type="seqinr::alignment")

# generate a distance matrix using seqinr package
d <- seqinr::dist.alignment(myAln2, "identity") %>% as.matrix() #computes distance as the square root of 1-similarity
d <- (1-(d^2))
tree <- nj(d)

#print order by unifrac distance
temp <- as.dendrogram(hclust(d = dist(x = d)))
otu.dendro <- order.dendrogram(temp)
factor(x = myAln2$nam, levels = myAln2$nam[otu.dendro], ordered = TRUE) %>% write.csv("../misc/otu_order_unifrac.csv")


#Combine all tables and plot
library("phyloseq")
set.seed(123)
OTU = otu_table(tSNE_input, taxa_are_rows = TRUE)
TAX = tax_table(taxmat)
SAMP = sample_data(meta%>% column_to_rownames('sample'))
pseq <- phyloseq(OTU, TAX, tree, SAMP)
temp <- phyloseq::distance(pseq, method = "uunifrac", type = 'Sample')
iMDS  <- ordinate(pseq, "MDS", distance=temp)
plot_ordination(pseq, iMDS, color = 'mouse_type') + 
  labs(color = 'Vendor') +
  scale_color_manual(values = ven) +
  theme_cowplot() +
          theme(
              axis.text.y = element_blank(), 
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              axis.ticks.y = element_blank()
)

ggsave("../figs/fig1/1B_VendorUnifrac.png", width = 4.5, height = 2.5)
ggsave("../figs/fig1/1B_VendorUnifrac.pdf", width = 4.5, height = 2.5)

#get dat sample data
sampledf <- data.frame(sample_data(pseq))

# Adonis PERMANOVA test to show significance
permanova <- adonis(temp ~ mouse_type, data = sampledf)
permanova$aov.tab$`Pr(>F)`[1]
#.001!

```


Fig 1C - Shannon Diversity of 16S seqs
```{r}
#library(entropart)
library(vegan)
set.seed(123)

vendor <- zotu %>% filter(day == 0, OTU != 'Otu10') %>% #Look @ D0 samples, remove sporo spike-in
  select(OTU, seqid,mouse_type, relabd) %>% filter( relabd > .005) %>%#mutate(relabd=log10(relabd)) %>%
  reshape2::dcast(., mouse_type + seqid ~ OTU) 

vendor[is.na(vendor)] <- 0

grp_key <- vendor[,1:2]

#create such that samples as columns, rows as features
tSNE_header <- (vendor %>% select(seqid) %>% t() %>% as.data.frame())[1,]


tSNE_input <- (vendor %>% select(-seqid) %>% t() %>% as.data.frame())[-1,] %>% 
              mutate_all(., function(x) as.numeric(as.character(x))) 

names(tSNE_input) <- tSNE_header[1,]


sd_df<- tSNE_input %>% t() %>% vegan::diversity(index = "shannon")
#sd_df<- tSNE_input %>% t() %>% vegan::tsallis(norm = T)
#sd_df <- tSNE_input %>% t() %>% entropart::PhyloEntropy(., q =1, pseq)

sd_lab <- tSNE_header %>% t()

shannonD <- sd_df %>% as.data.frame() %>% cbind(., sd_lab) %>% select(shannD = '.', seqid)

#normalize by log(k) where k is number of species over 0.05
numk <- zotu %>% filter(day == 0, OTU != 'Otu10') %>% select(OTU, seqid, relabd) %>% 
  melt(id.vars = c('seqid', 'OTU')) %>% 
  filter(value > .005) %>% 
  select(-value) %>% distinct() %>% 
  group_by(seqid) %>% mutate(kval = log2(n())) %>%
  ungroup() %>% select(seqid, kval) %>% distinct() %>% na.omit()

shannonD <- left_join(shannonD, numk) %>% mutate(shann_entropy = shannD/kval) %>% left_join(.,grp_key)

#IF ONLY DOING SHANNON DIVERSITY
#shannonD <- left_join(shannonD, numk) %>% left_join(.,grp_key)

#all side by side
shannonD$mouse_type <- factor(shannonD$mouse_type,      # Reordering group factor levels
levels = rev(c("Taconic", "Jackson", "Charles River","Envigo")))

ggplot(shannonD, aes(factor(mouse_type, levels = c("Taconic", "Jackson", "Charles River", "Envigo")), 
                     shann_entropy), fill = 'white') +
                     #shannD,)) + 
    stat_boxplot(geom= 'errorbar', width = .2, lwd = 1) +
  geom_boxplot(outlier.shape = NA, lwd = 1) + geom_jitter() +
 # scale_fill_manual(values = c('#FAAF40',"#1B75BB", "#F05A28","#37B34A")) +
  labs(y = 'Normalized Shannon Entropy', x = '') +
  theme_clean() +
  ggsignif::geom_signif(comparisons = list(c("Envigo", "Charles River"),
                                           c("Charles River", "Taconic"),
                                           c("Taconic", "Jackson"),
                                           c("Envigo", "Jackson")),
                         y_position = c(0.65, 0.63, 0.61, .67),
                         map_signif_level=TRUE, #Comment out for p-values
                         test = "wilcox.test") +
  theme(axis.text.x = element_text(size = 14), 
  axis.text.y = element_text(size = 14),
axis.title.x.bottom = element_text(size = 14),
axis.title.y.left = element_text(size = 14),
legend.position = 'none')

wilcox.test(filter(shannonD, mouse_type == 'Envigo')$shann_entropy, filter(shannonD, mouse_type == 'Charles River')$shann_entropy) #0.009
wilcox.test(filter(shannonD, mouse_type == 'Charles River')$shann_entropy, filter(shannonD, mouse_type == 'Taconic')$shann_entropy) #0.00001
wilcox.test(filter(shannonD, mouse_type == 'Taconic')$shann_entropy, filter(shannonD, mouse_type == 'Jackson')$shann_entropy) #0.0002

ggsave("../figs/fig1/1C_vendor_ShannonDiversity.png", width = 4.75, height = 3.5)
ggsave("../figs/fig1/1C_vendor_ShannonDiversity.pdf", width = 4.75, height = 3.5)

write.csv(shannonD, "../misc/VendorDiversity.csv", quote = F, row.names = F)

```



Supplementary Figure 1: 16S barplots for different vendors or relative amounts

```{r}

bugColors <- setNames(c(scales::hue_pal()(length(unique(vendor.long$family)))), c(levels(as.factor(vendor.long$family))))

write.table(bugColors, "../tables/bugColors2.txt", row.names = T, sep = '\t')

plot <- vendor.long %>% group_by(mouse_type, OTU) %>% 
    mutate(mean_log10relabd = mean(value)) %>% select(-value) %>% distinct() %>%
  ggplot(., aes(x = mouse_type, y=10^mean_log10relabd, fill = family)) + 
  geom_col(stat="identity", position = 'fill') +
  #geom_col(stat="identity", position = 'stack') +
  labs(x='', y = 'Rel. Abundance', fill = 'Family') +
    scale_fill_manual(values = bugColors) +
  theme_clean() +
    theme(plot.background = element_blank(),
          strip.background = element_blank(),
          axis.title.y = element_text(size = 14),
              axis.text.y = element_text(size = 12), 
              axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.ticks.x = element_blank(),
              panel.spacing = unit(0.05, "lines"),
          legend.text = element_text(size = 8),
          legend.key.height  = unit(.35, 'cm'),
          legend.key.width  = unit(.35, 'cm'))



#plotly::ggplotly(plot)
plot

plotb <- vendor.long %>% group_by(mouse_type, OTU) %>% 
  
  left_join(., (tax %>% select(genus, phylum) %>% distinct())) %>%
    mutate(mean_log10relabd = mean(value)) %>% select(-value) %>% distinct() %>%
  ggplot(., aes(x = mouse_type, y=10^mean_log10relabd, fill = phylum)) + 
  geom_col(stat="identity", position = 'fill') +
  #geom_col(stat="identity", position = 'stack') +
  labs(x='', y = 'Rel. Abundance', fill = 'Phylum') +
  #  scale_fill_manual(values = bugColors) +
  theme_clean() +
    theme(plot.background = element_blank(),
          strip.background = element_blank(),
          axis.title.y = element_text(size = 14),
              axis.text.y = element_text(size = 12), 
              axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
              axis.ticks.x = element_blank(),
              panel.spacing = unit(0.05, "lines"),
          legend.text = element_text(size = 8),
          legend.key.height  = unit(.35, 'cm'),
          legend.key.width  = unit(.35, 'cm'))

cowplot::plot_grid(plot,  plotb)

ggsave("../figs/s1/S1A_Vendor_barplots.png", width = 10, height = 4)
ggsave("../figs/s1/S1A_Vendor_barplots.pdf", width = 10, height = 4)

```


Looking at number of OTUs transferred?
```{}

zotu$mouse_type <- factor(zotu$mouse_type,      # Reordering group factor levels
                         levels = c("Jackson", "Taconic", "Envigo", "Charles River"))

zotu %>% filter(day == 0) %>% left_join(., gtable) %>%
    select(mouse_type, family, relabd, OTU) %>% 
    group_by(mouse_type, OTU) %>% 
      mutate(mean_log10relabd = mean(relabd)) %>% distinct() %>%
  filter(mean_log10relabd > 50) %>% select(-relabd, -OTU) %>% 
  distinct() %>%
  ggplot(., aes(x = mouse_type, fill = family), color = 'black') + 
  geom_bar(position = 'dodge', color = 'black') +
  labs(x='', y = '# OTU', fill = 'Family', title = 'Richness') +
  scale_fill_manual(values = bugColors) +
  theme_clean() +
    theme(plot.background = element_blank(),
          axis.title.y = element_text(size = 14),
              axis.text.y = element_text(size = 12), 
              axis.text.x = element_text(size = 12),
              axis.ticks.x = element_blank(),
          legend.key.height  = unit(.35, 'cm'),
          legend.key.width  = unit(.35, 'cm'))



```


