---
title: "Fig4_FMT"
output: html_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)

setwd("~/Dropbox/fmt/scripts/")
library(tidyverse)
library(wesanderson)
library(ggthemes)
library(reshape2)

ven <- c('#FAAF40',"#1B75BB","#37B34A", "#F05A28")
pal <- wes_palette("Zissou1", 100, type = "continuous")
#write.csv(bugColors, "../tables/bugColors.csv", quote = F, row.names = F)

bugColors <- read.csv("../tables/bugColors.csv")$x
  
options(stringsAsFactors = F)
options(scipen = 10000)

zotu <- read_csv("../seq/zotu.csv") 
tax <- read_csv("../seq/tax.csv")
meta <- read_csv("../experiments/2021-03-18_JxE_FMT03/metadata/seq_sac_samples.csv") %>% tail(., -12) #Remove old cecum samples with poor sequencing
        

#meta <- read_csv("~/Google_Drive/WangLab/Projects/FMT/March2021_FMT_recreate/LumenalContentsAnalysis/data/seq_sac_samples.csv")



```

```{r}

# Check read counts
read.counts <- zotu %>%
  rowwise(seqid) %>%
  mutate(numreads = rowSums(across(where(is.numeric)))) %>%
  ungroup() %>%
  select(seqid, numreads)

# Pull out samples w/ >10k reads
samples.to.keep <- read.counts %>%
  filter(numreads > 1000) %>%
  select(seqid) %>%
  as_vector()

# Drop <1k read samples from zotu table
zotu <- zotu %>%
  filter(seqid %in% samples.to.keep)

# make tidy data
# Convert to relative abundance

zotu <- zotu %>%
  pivot_longer(!seqid, names_to="OTU", values_to="reads") %>%
  group_by(seqid) %>%
  mutate(reads = reads/sum(reads)) %>%  #normalize to number of reads per sample, before was scaled * 3x10^4
  inner_join(meta, ., by="seqid")


#Dataframe with normalization values
normalizer <- zotu %>% filter(OTU == "Otu1") %>% 
                      mutate(norm_factor_sporo = (reads/min(reads)),   #calculate normalization factor for sporo
                      norm_factor_pellet = (sample_weight/min(sample_weight))) %>% #calculate normalization factor for pellet
                      select(seqid, norm_factor_sporo, norm_factor_pellet)

#shared scaling factors so values appear somewhat normal
med_sporo <- median(normalizer$norm_factor_sporo) 
med_sample_weight <- median(normalizer$norm_factor_pellet)

norm_zotu <- inner_join(zotu, normalizer, by = 'seqid') %>% 
        mutate(abs_reads = ifelse(reads==0, 0, 
                (((reads/norm_factor_sporo)*med_sporo)/norm_factor_pellet)*med_sample_weight)) %>% #normalize here
      filter(OTU != 'Otu1') %>%   #Remove sporosarcina going forward
      group_by(seqid) %>%
      mutate(abs_rel_abundance = abs_reads/sum(abs_reads)) %>% ungroup()


#quick renames
#norm_zotu[norm_zotu == "envigo_control"] <- "Envigo Control"
#norm_zotu[norm_zotu == "jax_control"] <- "Jax Control"
#norm_zotu[norm_zotu == "jax_envigo_mix"] <- "Jax Cohoused"

norm_zotu[norm_zotu == "envigo_control"] <- "Donor"
norm_zotu[norm_zotu == "jax_control"] <- "- FMT"
norm_zotu[norm_zotu == "jax_envigo_mix"] <- "+ FMT"

```


Filter out low abundance OTU, assign taxonomical information
```{r}

temp <- norm_zotu

#Only look at microves with .5% abundance in any location, any sample
otus.to.keep <- norm_zotu %>%
  group_by(OTU, sample_location) %>%
  summarise(avg_abd = max(abs_rel_abundance)) %>%
  filter(avg_abd > .02) %>%
  select(OTU) %>% distinct() %>%
  as_vector() 


# Drop OTUs not at least .5% abd
norm_zotu <- norm_zotu %>%
  filter(OTU %in% otus.to.keep)

# calc log10 OTU abd
norm_zotu <- norm_zotu %>%
  mutate(log10abs_rel_abundance = if_else(abs_rel_abundance == 0, -5, log10(abs_rel_abundance)),
         log10absl_abundance = if_else(abs_reads == 0, -5, log10(abs_reads)))


gtable <- tax %>%
  select(OTU, genus, family)

norm_zotu <- norm_zotu %>%
  left_join(gtable, by="OTU") %>%
  mutate(otu_genus = paste(OTU, genus, sep="_"))

```

Fig 4A - Spatial 16S abundance
```{r}

library('ggdendro')
library('grid')

set.seed(123)

jax.cohoused <- norm_zotu %>% filter(tx_group == '+ FMT') %>%
    group_by(sample_location, OTU, mouse_type) %>% filter(OTU != 'Otu1') %>% 
    summarise(mean_log10abs_rel_abundance = mean(log10abs_rel_abundance)) %>% #can change to abs_rel_abundance to normalize for location
    ungroup() %>%
    select(OTU, sample_location, mean_log10abs_rel_abundance) %>% distinct() %>% reshape2::dcast(OTU~sample_location)


jax.cohoused.scaled <- jax.cohoused
jax.cohoused.scaled[, c(2:6)] <- (jax.cohoused.scaled[, 2:6]) #don't scale here actually

#clustering
jax.cohoused.matrix <- as.matrix(jax.cohoused.scaled[, -c(1)])
rownames(jax.cohoused.matrix) <- jax.cohoused.scaled$OTU
jax.cohoused.dendro <- as.dendrogram(hclust(d = dist(x = jax.cohoused.matrix)))

# Create dendro
dendro.plot <- ggdendrogram(data = jax.cohoused.dendro, rotate = TRUE)

# Preview the plot
print(dendro.plot)


#reorder
jax.cohoused.order <- order.dendrogram(jax.cohoused.dendro)
#jax.cohoused.long <- left_join(jax.cohoused.long, gtable)
norm_zotu$OTU <- factor(x = norm_zotu$OTU,
                               levels = jax.cohoused.scaled$OTU[jax.cohoused.order], 
                               ordered = TRUE)

bugColors <- setNames(c(scales::hue_pal()(length(unique(norm_zotu$family)))), c(levels(as.factor(norm_zotu$family))))

norm_zotu$tx_group <- factor(norm_zotu$tx_group,      # Reordering group factor levels
                         levels = c("- FMT", "+ FMT", "Donor"))

#fam lvl
heatmap.plot <- norm_zotu %>%
  group_by(sample_location, tx_group, family, OTU) %>%
  summarise(mean_log10absl_abundance = mean(log10abs_rel_abundance)) %>%
  group_by(tx_group, sample_location) %>% 
  #mutate(mean_log10abs_rel_abundance = log10(10^mean_log10abs_rel_abundance/sum(10^mean_log10abs_rel_abundance))) %>%
  ggplot() +
  geom_tile(aes(x=factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), 
                y=OTU, fill=mean_log10absl_abundance), color = 'white') +
  #scale_fill_gradientn(colours = pal) + 
  scale_fill_gradient2(high = 'firebrick3', mid = 'white', low = '#1357a6', midpoint = -3) +
  labs(x = 'Location', y = 'OTU', fill = 'Rel. Abundance (log10)') +
  facet_wrap(~tx_group) +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              axis.text.y = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'bottom',
              panel.spacing = unit(0.05, "lines"))
heatmap.plot
fam <- ggplot(data = norm_zotu, aes(x = '', y = OTU)) +
    geom_tile(aes(fill = family)) + 
    labs(fill = 'Family') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_blank(), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))

cowplot::plot_grid(heatmap.plot, fam, ncol =2, rel_widths = c(.6,.4))

ggsave("../figs/fig4/Spatial_FMT.png",  width = 10, height = 5)
ggsave("../figs/fig4/Spatial_FMT.pdf",  width = 10, height = 5)

```


Are OTUs redistributed across gut?
```{r}
colon_levels <- c("duodenum","jejunum", "ileum","cecum", 'colon')

#Bacteroides
temp <- norm_zotu %>%
  group_by(sample_location, tx_group, family, OTU) %>%
  summarise(mean_log10abs_rel_abundance = mean(log10absl_abundance)) %>% ungroup() %>% distinct() %>%
  filter(OTU == 'Otu4' & tx_group != "Donor")

 a <- ggplot(filter(temp, tx_group == "+ FMT"), 
         aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    geom_point(data = temp, aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) +
    geom_line() + 
    cowplot::theme_cowplot() + 
    geom_line(data=filter(temp, tx_group == "- FMT"), 
              aes(x=factor(sample_location, levels=colon_levels),
                  y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    scale_color_manual(values = c('#ffa500','#00ba38')) + 
    labs(x = 'Location', y='Abs. Abundance (log10)', color = 'Cohort', 
         title = "OTU4: Duncaniella") + 
    theme(plot.title = element_text(hjust = 0.5, size = 8), 
          legend.position = 'none',
          axis.text.x = element_blank(),
          axis.title.y = element_text(size = 10))
  

#Lactobacillae
temp <- norm_zotu %>%
  group_by(sample_location, tx_group, family, OTU) %>%
  summarise(mean_log10abs_rel_abundance = mean(log10absl_abundance)) %>% ungroup() %>% distinct() %>%
  filter(OTU == 'Otu10' & tx_group != "Donor")

 b <- ggplot(filter(temp, tx_group == "+ FMT"), 
         aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    geom_point(data = temp, aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) +
    geom_line() + 
    cowplot::theme_cowplot() + 
    geom_line(data=filter(temp, tx_group == "- FMT"), 
              aes(x=factor(sample_location, levels=colon_levels),
                  y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    scale_color_manual(values = c('#ffa500','#00ba38')) + 
    labs(x = 'Location', y='Abs. Abundance (log10)', color = 'Cohort', 
         title = "OTU10: Lactobacillus") + 
    theme(plot.title = element_text(hjust = 0.5, size = 8), 
          legend.position = 'none',
          axis.text.x = element_blank(),
          axis.title.y = element_blank())
 
 #Bifidobacteriaceae
temp <- norm_zotu %>%
  group_by(sample_location, tx_group, family, OTU) %>%
  summarise(mean_log10abs_rel_abundance = mean(log10absl_abundance)) %>% ungroup() %>% distinct() %>%
  filter(OTU == 'Otu138' & tx_group != "Donor")

 c <- ggplot(filter(temp, tx_group == "+ FMT"), 
         aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    geom_point(data = temp, aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) +
    geom_line() + 
    cowplot::theme_cowplot() + 
    geom_line(data=filter(temp, tx_group == "- FMT"), 
              aes(x=factor(sample_location, levels=colon_levels),
                  y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    scale_color_manual(values = c('#ffa500','#00ba38')) + 
    labs(x = 'Location', y='Abs. Abundance (log10)', color = 'Cohort', 
         title = "OTU138: Bifidobacterium") + 
    theme(plot.title = element_text(hjust = 0.5, size = 8), 
          legend.position = 'none', 
          axis.text.x = element_blank(),
          axis.title.y = element_blank())
 
 
  #Lactobacillae
temp <- norm_zotu %>%
  group_by(sample_location, tx_group, family, OTU) %>%
  summarise(mean_log10abs_rel_abundance = mean(log10absl_abundance)) %>% 
  ungroup() %>% distinct() %>%
  filter(OTU == 'Otu52' & tx_group != "Donor")

 d <- ggplot(filter(temp, tx_group == "+ FMT"), 
         aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    geom_point(data = temp, aes(x=factor(sample_location, levels=colon_levels),
             y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) +
    geom_line() + 
    cowplot::theme_cowplot() + 
    geom_line(data=filter(temp, tx_group == "- FMT"), 
              aes(x=factor(sample_location, levels=colon_levels),
                  y=mean_log10abs_rel_abundance, group = 1, color = tx_group)) + 
    scale_color_manual(values = c('#ffa500','#00ba38')) + 
    labs(x = 'Location', y='Abs. Abundance (log10)', color = 'Cohort', 
         title = "OTU52: Turicibacter") + 
    theme(plot.title = element_text(hjust = 0.5, size = 8),
          axis.text.x = element_blank(),
          axis.title.y = element_blank())

cowplot::plot_grid(a,b,c,d, rel_widths = c(1,1,1,1.6), ncol = 4)
ggsave("../figs/fig4/4B_RedistributionOTUs.png", height = 2, width = 8)
ggsave("../figs/fig4/4B_RedistributionOTUs.pdf", height = 2, width = 8)

```



How is absolute abundance across different areas by vendor?
```{r}

p <- norm_zotu %>%
  group_by(sample_location, mouse_num) %>%
  mutate(sum_abs_reads = sum(abs_reads)) %>% ungroup() %>% 
  select(sum_abs_reads, sample_location, tx_group) %>% distinct() %>%
  ggplot(., aes(x=factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')),
                y=log10(sum_abs_reads),
                fill=tx_group)) + 
  geom_boxplot(outlier.shape = NA) +
  geom_point(pch = 21, position = position_jitterdodge()) +
  scale_fill_manual(values = c('#f7931e','#d9e021','#29abe2')) +
  #scale_fill_manual(values = c('#37b34a', '#7e09d5', '#1b75bb')) +
 # scale_fill_manual(values = c('lightblue', 'blue', 'green')) +
 # scale_y_log10() +
  labs(x = '', y = 'Rel. Biomass (log10)', fill = 'Cohort') +
  theme_clean() +
  theme(axis.text.x = element_text(size = 12), 
        axis.text.y = element_text(size = 12),
        axis.title.x.bottom = element_text(size = 12),
        axis.title.y.left = element_text(size = 12)) 
p

#median biomass changes from 10^3.99 > 10^2.44 in the duodenum (97.2% decrease)

ggsave("../figs/fig4/4D_microbiomeAbd_FMT.png",  width = 6, height = 3)
ggsave("../figs/fig4/4D_microbiomeAbd_FMT.pdf",  width = 6, height = 3)

```





4C What is the abundance of Donor vs Recipient vs shared taxa across the gut

```{r}

jax.otus.to.keep <- norm_zotu %>%
  filter(tx_group == "- FMT") %>%
  group_by(OTU, sample_location) %>%
  summarise(avg_abd = max(abs_rel_abundance)) %>%
  filter(avg_abd > .01) %>%
  select(OTU) %>% distinct() %>%
  as_vector() 

envigo.otus.to.keep <- norm_zotu %>%
  filter(tx_group == "Donor") %>%
  group_by(OTU, sample_location) %>%
  summarise(avg_abd = max(abs_rel_abundance)) %>%
  filter(avg_abd > .01) %>%
  select(OTU) %>% distinct() %>%
  as_vector() 


otus.jax.Specific <- setdiff(jax.otus.to.keep, envigo.otus.to.keep) %>%
                  cbind('Recipient Specific') %>% as.data.frame() 

names(otus.jax.Specific) <- c("OTU", "class")

otus.envigo.Specific <- setdiff(envigo.otus.to.keep, jax.otus.to.keep) %>% 
  cbind('Donor Specific') %>% as.data.frame() 

names(otus.envigo.Specific) <- c("OTU", "class")

otus.shared.Specific <- intersect(jax.otus.to.keep, envigo.otus.to.keep) %>%
    cbind('Shared') %>% as.data.frame()

names(otus.shared.Specific) <- c("OTU", "class")


otu_labels <- rbind(otus.envigo.Specific, otus.jax.Specific, otus.shared.Specific)
temp <- norm_zotu %>% 
    group_by(sample_location, OTU, tx_group) %>% filter(OTU != 'Otu1') %>% 
    summarise(mean_abs = mean(abs_reads)) %>% 
  select(OTU, sample_location, mean_abs, tx_group) %>%
  left_join(.,otu_labels) %>% group_by(class, sample_location, tx_group) %>% 
        mutate(class_abd = sum(mean_abs)) %>%
        ungroup() %>%
        select(sample_location, class, class_abd, tx_group) %>% distinct()

ggplot(temp, aes(x=factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')),
                y=class_abd,
                fill=factor(class, levels = c('Donor Specific', 'Shared', 'Recipient Specific')))) +  
  geom_bar(stat="identity", position = 'fill') +
  facet_wrap(~tx_group, ncol = 3, strip.position = 'top') +
  labs(x='Sample Location', y = 'Proportion of Microbiome', fill = 'Microbe Origin') +
  #scale_fill_manual(values = c(ven[3], 'gray80', ven[2])) +
  scale_fill_manual(values = c(ven[2], 'gray90', '#ffa500')) +
  theme_clean() +
    theme(plot.background = element_blank(),
          strip.background = element_blank(),
          axis.title.y = element_text(size = 14),
          axis.title.x = element_text(size = 14),
              axis.text.y = element_text(size = 12),
              axis.text.x = element_text(size = 12, hjust = 1, angle = 45), 
              strip.text = element_text(size = 12),
              panel.spacing = unit(0.05, "lines"),
          legend.position = 'top')


ggsave("../figs/fig4/4E_microbiomeClass_FMT.png",  width = 6, height = 3.5)
ggsave("../figs/fig4/4E_microbiomeClass_FMT.pdf",  width = 6, height = 3.5)

```


4B) Colonizers by location
```{r}

ctrl <- norm_zotu %>% filter(tx_group == '- FMT') %>% 
  group_by(OTU, sample_location) %>% 
  mutate(ctrl_abd = mean(abs_reads)) %>% 
  select(OTU, sample_location, ctrl_abd) %>% ungroup() %>% distinct()

fmt <- norm_zotu %>% filter(tx_group == '+ FMT') %>% 
  group_by(OTU, sample_location) %>% 
  mutate(fmt_abd = mean(abs_reads)) %>% 
  select(OTU, sample_location, fmt_abd) %>% ungroup() %>% distinct()

colonizing_location <- full_join(ctrl, fmt) %>% 
          mutate(delta_abd = (fmt_abd-ctrl_abd),
          colonizer_type = ifelse(delta_abd > 0, 'Gained', 'non_colonizer'),
          colonizer_type = ifelse(delta_abd < 0, 'Lost', colonizer_type)) %>%
          group_by(sample_location) %>% mutate(delta_abd_norm=delta_abd/sum(ctrl_abd)) %>% 
          ungroup() %>% left_join(., gtable)


a <- colonizing_location %>% filter(colonizer_type == 'Gained') %>% 
  ggplot(., aes(x = factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), 
                y=(delta_abd_norm),
                fill = family)) + 
      scale_fill_manual(values = bugColors) +
    #ylim(0,42000) +
  geom_bar(stat="identity", position = 'stack') +
  labs(x='', y = 'Relative Biomass', fill = 'Family', title = 'Gained') +
  theme_clean() +
    theme(
      plot.background = element_blank(),
          strip.background = element_blank(),
          title = element_text(size = 10),
          axis.title.y = element_text(size = 12),
          axis.title.x = element_text(size = 12),
              axis.text.y = element_text(size = 8),
              axis.text.x = element_text(size = 8,  hjust = 1, angle = 45), 
              strip.text = element_text(size = 8),
              panel.spacing = unit(0.05, "lines"),
      plot.title = element_text(hjust = 0.5),
          legend.position = 'none', 
          )


b <- colonizing_location %>% filter(colonizer_type == 'Lost') %>%
  ggplot(., aes(x = factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), 
                y=(abs(delta_abd_norm)),
                fill = family)) + 
      scale_fill_manual(values = bugColors) +
   #ylim(0,42000) +
  geom_bar(stat="identity", position = 'stack') +
  labs(x='', y = 'Relative Biomass', fill = 'Family', title = 'Displaced') +
  theme_clean() +
    theme(
      plot.background = element_blank(),
          strip.background = element_blank(),
          title = element_text(size = 10),
          axis.title.y = element_text(size = 12),
          axis.title.x = element_text(size = 12),
              axis.text.y = element_text(size = 8),
              axis.text.x = element_text(size = 8,  hjust = 1, angle = 45), 
              strip.text = element_text(size = 8),
              panel.spacing = unit(0.05, "lines"),
      plot.title = element_text(hjust = 0.5),
      legend.position = 'none')

cowplot::plot_grid(a,b,nrow = 1)

#plotly::ggplotly(a)

ggsave("../figs/fig4/4C_Colonization_Spatial.pdf",  width = 4, height = 2.5)


legend <- colonizing_location %>% filter(colonizer_type == 'Lost') %>%
              ggplot(., aes(x = sample_location, 
                y=abs(delta_abd),
                fill = family)) +
  labs(fill = 'Family') +
    geom_bar(stat="identity", position = 'stack') +
      scale_fill_manual(values = bugColors) + theme(legend.position = 'right')

legend <- cowplot::get_legend(legend)
grid.newpage()

grid.draw(legend)

ggsave("../figs/fig4/4C_legend.pdf",  width = 10, height = 1.5)


```


S4A - How do gut compartments localize now? PCA euclidean distance by absolute # of reads
```{r}

library(M3C)

compartments <- norm_zotu %>% arrange(tx_group) %>% 
  arrange(factor(sample_location, levels = c('duodenum', 'jejunum', 'ileum', 'cecum', 'colon'))) %>%
  select(OTU, seqid, log10absl_abundance) %>% 
  reshape2::dcast(seqid~OTU, fun.aggregate=mean) 


compartments.matrix <- as.matrix(compartments[, -c(1)])

grp_key <- norm_zotu %>% select(seqid, tx_group, sample_location)

#create such that samples as columns, rows as features
#tSNE_header <- (compartments %>% t() %>% as.data.frame())[1,]
tSNE_header <- (compartments %>% select(seqid) %>% left_join(., grp_key) %>% distinct() %>%
    mutate(header = paste(tx_group, sample_location, sep = '_'))  %>% 
    select(header) %>% t() %>%
  as.data.frame())[1,]


tSNE_input <- (compartments %>% t() %>% as.data.frame())[-1,] %>% 
              mutate_all(., function(x) as.numeric(as.character(x))) 

names(tSNE_input) <- tSNE_header[1,]

colores <- c('darkorange3','darkorange4','darkorange1','darkorange3','darkorange2', 
             'gold3', 'gold4', 'gold1', 'gold3', 'gold2',
             'skyblue3','skyblue4','skyblue1','skyblue3','skyblue2')
#Plot PCA
set.seed(123)
s9c <- M3C::pca(tSNE_input, 
        labels = (names(tSNE_input)),
         colvec = colores,
         dotsize = 3,
         controlscale = T, scale = 3, 
         axistextsize = 10,
         legendtextsize = 14, legendtitle = '')

plotly::ggplotly(s9c)

ggsave("../figs/s4/S4A_PCA_Gutcompartments.png", width = 6, height = 3.5)
ggsave("../figs/s4/S4A_PCA_Gutcompartments.pdf", width = 6, height = 3.5)


#Look at within-group dissimilarity to see whether jax control gut compartments are less dissimilar
sampledf <- data.frame(sample_data(pseq))

gut.ano <- with(sampledf, anosim(temp, tx_group))
summary(gut.ano)
plot(gut.ano)
autoplot(gut.ano, notch = FALSE) + geom_jitter(alpha = .2) +# scale_x_discrete(breaks = c('envigo_control', 'jax_control', 'jax_envigo_mix')) +
    ggsignif::geom_signif(
    comparisons = list(c('envigo_control', 'jax_control'), c('jax_control', 'jax_envigo_mix')),
    test = 'wilcox.test',
    y_position = c(1350, 1200)
  ) + theme_cowplot() + labs(y = 'Distance Rank', x = 'Mouse Group', title = '') + 
  theme(axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12, angle = 45, hjust = 1),
        axis.title = element_text(size = 14))

ggsave("../figs/s4/S4B_anosim_Gutcompartments.png", width = 3.5, height = 4.5)
ggsave("../figs/s4/S4B_anosim_Gutcompartments.pdf", width = 3.5, height = 4.5)


```


S4C - Are late/early colonizers colonizing specific areas?



12/17/23 - Needed to fix this since it didn't properly normalize for proportion of biomass. before it jsut had absolute reads at each spot (which were normalized across samples). now we normaize reads to total in gut compartment

```{r}

norm_zotu %>% inner_join(., read.csv("../tables/colonizer_order.txt")) %>%
  filter(tx_group == '+ FMT') %>%
  group_by(sample_location, tx_group, OTU) %>%
  mutate(mean_log10abs_rel_abundance = mean(abs_rel_abundance)) %>%
  select(family, OTU, mean_log10abs_rel_abundance, sample_location, colonizer_type) %>% 
  distinct() %>% ungroup() %>%
  group_by(sample_location, tx_group) %>% mutate(location_abd = sum(mean_log10abs_rel_abundance)) %>% ungroup %>%
  group_by(sample_location, tx_group, family, colonizer_type) %>%
      mutate(famabd = sum(mean_log10abs_rel_abundance),
             rel_loc = famabd/location_abd) %>%
  ungroup %>% select(sample_location, famabd,rel_loc, colonizer_type, family) %>% distinct() %>% 
  filter(colonizer_type != 'Unstable') %>%
  ggplot() +
  geom_bar(aes(x=factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), 
               # y=famabd, fill=family), stat = 'identity', width = .5, color = 'black') +
                 y=rel_loc, fill=family), stat = 'identity', width = .5, color = 'black') +
        scale_fill_manual(values = bugColors) +
  facet_wrap(~colonizer_type, scales = 'free_y', nrow = 1) +
  labs(x = 'Location', y = 'Proportion of Biomass \nat GI site', fill = 'Family') +
  cowplot::theme_cowplot() +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              axis.text.y = element_text(size = 10),
              axis.title.y = element_text(size = 12),
              axis.title.x = element_text(size = 12),
              strip.text = element_text(size = 10),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'right',
              panel.spacing = unit(0.05, "lines"))

ggsave("../figs/s4/S4C_Spatial_colonization_FMT.png",  width = 7.5, height = 3)
ggsave("../figs/s4/S4C_Spatial_colonization_FMT.pdf",  width = 7.5, height = 3)


```

OLD. S4B - Are these significantly different?
```{}
#source("https://bioconductor.org/biocLite.R")
#biocLite("qvalue")
#BiocManager::install("qvalue")
#library(qvalue)

#Get label grid
sample_levels <- (expand.grid(c('- FMT', '+ FMT', 'Donor'), c("duodenum","jejunum", "ileum","cecum", 'colon')) %>% 
  arrange(factor(Var1, levels = c('- FMT', '+ FMT', 'Donor'))) %>% 
    mutate(labels = paste(Var1, Var2, sep = '_')))$labels

#Get test frame to expand
testframe <- norm_zotu %>% arrange(tx_group) %>% 
  arrange(factor(sample_location, levels = c('duodenum', 'jejunum', 'ileum', 'cecum', 'colon'))) %>%
  mutate(testgroup = paste(tx_group, sample_location, sep = '_')) %>%
  select(OTU, testgroup, log10absl_abundance) 

temp <- pairwise.wilcox.test(testframe$log10absl_abundance, testframe$testgroup, p.adjust = "BH")$p.value %>% 
  as.data.frame() %>% mutate(yfact=as.factor(row.names(.))) %>% reshape2::melt(id.vars='yfact') %>%
  mutate(variable = as.factor(variable)) %>% na.omit() %>%
  rbind(., select(., yfact=variable, variable=yfact, value)) %>% distinct() %>% mutate(significant = ifelse(value < .01, TRUE, FALSE)) 


temp %>% ggplot(., aes(x=factor(variable, levels = sample_levels), 
                       factor(yfact, levels = rev(sample_levels)), fill = log10(value))) + 
  geom_tile(color = 'white') + theme_clean() +  
  scale_fill_gradient2(low = 'firebrick3', mid = 'white', high = '#1357a6', midpoint = -30) +
  geom_point(data=(temp %>% filter(significant == TRUE)), aes(variable, yfact), size=.75,alpha=.7,color="black",shape=8) +
  labs(fill = 'log10(adjst. p-value)', x = 'Env              Env2Jax            Jax', y = 'Env    Env2Jax         Jax     ') + 
  theme(axis.text.x = element_text(size = 12, hjust = 1, angle = 45),
              axis.text.y = element_text(size = 12),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14))


ggsave("../figs/s4/S4B_pairwise_Gutcompartments.png", width = 8, height = 5)
ggsave("../figs/s4/S4B_pairwise_Gutcompartments.pdf", width = 8, height = 5)

```


Doesn't work! Let's try jaccard similarity
```{}
#source("https://bioconductor.org/biocLite.R")
#biocLite("qvalue")
#BiocManager::install("qvalue")
#library(qvalue)

#Get label grid
sample_levels <- (expand.grid(c('- FMT', '+ FMT', 'Donor'), c("duodenum","jejunum", "ileum","cecum", 'colon')) %>% 
  arrange(factor(Var1, levels = c('- FMT', '+ FMT', 'Donor'))) %>% 
    mutate(labels = paste(Var1, Var2, sep = '_')))$labels

#Get test frame to expand
testframe <- norm_zotu %>% arrange(tx_group) %>% 
  arrange(factor(sample_location, levels = c('duodenum', 'jejunum', 'ileum', 'cecum', 'colon'))) %>%
  mutate(testgroup = paste(tx_group, sample_location, sep = '_')) %>%
  select(OTU, testgroup, log10absl_abundance) 

jaccard_input <- compartments.matrix
jaccard_input[compartments.matrix > -1] <- TRUE
jaccard_input[compartments.matrix < -1] <- FALSE

temp <- jaccard::jaccard.test.pairwise(jaccard_input)
  
  as.data.frame() %>% mutate(yfact=as.factor(row.names(.))) %>% reshape2::melt(id.vars='yfact') %>%
  mutate(variable = as.factor(variable)) %>% na.omit() %>%
  rbind(., select(., yfact=variable, variable=yfact, value)) %>% distinct() %>% mutate(significant = ifelse(value < .01, TRUE, FALSE)) 


temp %>% ggplot(., aes(x=factor(variable, levels = sample_levels), 
                       factor(yfact, levels = rev(sample_levels)), fill = log10(value))) + 
  geom_tile(color = 'white') + theme_clean() +  
  scale_fill_gradient2(low = 'firebrick3', mid = 'white', high = '#1357a6', midpoint = -30) +
  geom_point(data=(temp %>% filter(significant == TRUE)), aes(variable, yfact), size=.75,alpha=.7,color="black",shape=8) +
  labs(fill = 'log10(adjst. p-value)', x = 'Env              Env2Jax            Jax', y = 'Env    Env2Jax         Jax     ') + 
  theme(axis.text.x = element_text(size = 12, hjust = 1, angle = 45),
              axis.text.y = element_text(size = 12),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14))



```




```{}
#
heatmap.plot <- norm_zotu %>% 
  inner_join(., rbind(early_colonizers, late_colonizers)) %>% 
  filter(tx_group == '+ FMT') %>%
  group_by(sample_location, tx_group, OTU) %>%
  mutate(mean_log10abs_rel_abundance = mean(log10absl_abundance)) %>%
  ggplot() +
  geom_tile(aes(x=factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), 
                y=OTU, fill=mean_log10abs_rel_abundance)) +
  scale_fill_gradientn(colours = pal, limits = c(-2,4)) +
  facet_wrap(~type, scales = 'free') +
  labs(x = 'Location', y = 'OTU', fill = 'log10(Relative Abundance)', title = 'Early Colonizers') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
             # axis.text.y = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'bottom',
              panel.spacing = unit(0.05, "lines"))

fam <-norm_zotu %>% 
    inner_join(., rbind(early_colonizers, late_colonizers)) %>% 
    ggplot(., aes(x = '', y = OTU)) +
    geom_tile(aes(fill = family)) + 
    facet_wrap(~type, scales = 'free') +
    labs(fill = 'Family') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_blank(), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))

cowplot::plot_grid(heatmap.plot,fam, ncol =2, rel_widths = c(.6,.4))

ggsave("../figs/s4/S4A_Spatial_colonization_FMT.png",  width = 10, height = 5)
ggsave("../figs/s4/S4A_Spatial_colonization_FMT.pdf",  width = 10, height = 5)


heatmap.plot_late <- norm_zotu %>% 
  semi_join(., late_colonizers) %>% 
  filter(tx_group == '+ FMT') %>%
  group_by(sample_location, tx_group, OTU) %>%
  summarise(mean_log10abs_rel_abundance = mean(log10absl_abundance)) %>%
  ggplot() +
  geom_tile(aes(x=factor(sample_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), 
                y=OTU, fill=mean_log10abs_rel_abundance)) +
  scale_fill_gradientn(colours = pal, limits = c(-2,4)) + 
  labs(x = 'Location', y = 'OTU', fill = 'log10(Relative Abundance)', title = 'Late Colonizers') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              #axis.text.y = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'bottom',
              panel.spacing = unit(0.05, "lines"))

fam_late <-norm_zotu %>% 
    semi_join(., late_colonizers) %>% 
    ggplot(., aes(x = '', y = OTU)) +
    geom_tile(aes(fill = family)) + 
    labs(fill = 'Family') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_blank(), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))



cowplot::plot_grid(heatmap.plot_early,fam_early, heatmap.plot_late, fam_late, ncol =3, rel_widths = c(.6,.4,.6,.4))

ggsave("../figs/s4/S4A_Spatial_colonization_FMT.png",  width = 10, height = 5)
ggsave("../figs/s4/S4A_Spatial_colonization_FMT.pdf",  width = 10, height = 5)

```

