---
title: "fig6_fmt"
output: html_document
---

```{r setup, include=FALSE}

setwd("~/Dropbox/fmt/scripts/")

knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(wesanderson)
library(ggthemes)
library(reshape2)
library(cowplot)

ven <- c('#FAAF40',"#1B75BB","#37B34A", "#F05A28")
pal <- wes_palette("Zissou1", 100, type = "continuous")


options(stringsAsFactors = F)
options(scipen = 10000)

zotu <- read.table("../experiments/2020-12-20_miceFMT_master/formatted_tables/otu_table_zotu.txt", header = T)  %>% t() %>% as.data.frame() %>% mutate(seqid = row.names(.))
tax <- read.csv("../experiments/2020-12-20_miceFMT_master/formatted_tables/tax_HFMT.csv")
meta <- read.csv("../experiments/2020-12-20_miceFMT_master/formatted_tables/metadata_HFMT.csv") %>% 
  left_join(., read.csv("../experiments/2020-12-20_miceFMT_master/formatted_tables/days_HFMT.csv")) %>% 
  mutate(seqid = paste(day,mouse_num, sep = '')) %>% select(-join_hack) %>% 
  rbind(., read.csv("../experiments/2020-12-20_miceFMT_master/formatted_tables/metadata_HFMT_donors.csv")) #add human samples

```

```{r}
# Check read counts for samples
read.counts <- zotu %>%
  rowwise(seqid) %>%
  mutate(numreads = rowSums(across(where(is.numeric)))) %>%
  ungroup() %>%
  select(seqid, numreads)

# Pull out samples w/ >10k reads
samples.to.keep <- read.counts %>%
  filter(numreads > 10000) %>%
  select(seqid) %>%
  as_vector()

# Drop <1k read samples from zotu table
zotu <- zotu %>%
  filter(seqid %in% samples.to.keep)

# make tidy data
# Convert to relative abundance

zotu <- zotu %>%
  pivot_longer(!seqid, names_to="OTU", values_to="reads") %>%
  group_by(seqid) %>%
  #mutate(relabd = reads/sum(reads)*30000) %>%  #normalize to number of reads per sample
  mutate(relabd = reads/sum(reads)) %>%  #normalize to number of reads per sample
  mutate(log10relabd = if_else(relabd == 0, -6, log10(relabd))) %>% #calculate log10relabd
  inner_join(meta, ., by="seqid") 


```

Look @ OTUs to keep for human + mice transfer analysis
```{r}
temp_zotu <- zotu
#zotu <- temp_zotu

#Only look at microves with .5% abundance in any any sample
otus.to.keep <- zotu %>% #filter(species == 'human') %>%
  group_by(OTU, humanized_type) %>%
  summarise(avg_abd = max(relabd)) %>%
  filter(avg_abd > .001) %>%
  select(OTU) %>% distinct() %>%
  as_vector() 


# Drop OTUs filtered in previous step
zotu <- zotu %>%
  filter(OTU %in% otus.to.keep)

#Append taxonomic information
gtable <- tax %>%
  select(OTU, genus, family, order)

zotu <- zotu %>%
  left_join(gtable, by="OTU") %>%
  mutate(otu_genus = paste(OTU, genus, sep="_"))

```



Alternative to s9a, bar plot of family profiles
```{r}

bugColors_fam <- setNames(c(scales::hue_pal()(length(unique(zotu$family)))), c(levels(as.factor(zotu$family))))

#ave_h3 <- zotu %>% select(seqid == 'H3fec' | seqid = 'H3T2)
  
s9a2 <- zotu %>% filter(day == 'Day9' | species == 'human' ) %>% filter(seqid != 'H3T2fec') %>%
  filter(group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL' | species == 'human') %>%
  group_by(seqid, family) %>% mutate(fam_abd = sum(relabd)) %>% ungroup() %>% #select(-relabd, -OTU) %>%
  select(seqid, humanized_type, species, fam_abd, family) %>% distinct() %>%
  ggplot(., aes(x=seqid, y=fam_abd, fill=family), color = 'black') +
  geom_bar(stat = 'identity', position = 'fill') +
  #scale_fill_manual(values = bugColors_fam) + 
  labs(x = '', y = 'OTU', fill = 'log10(Relative Abundance)') +
  facet_grid(~humanized_type+species, scales = 'free') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              panel.grid = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'right',
              panel.spacing = unit(0.05, "lines"))

plotly::ggplotly(s9a2)
ggsave("../figs/s8/s8a_HFMT_sd.pdf", width = 8.5, height = 3.5)


```

Alternative to s9a, bar plot of family profiles
```{r}

bugColors_fam <- setNames(c(scales::hue_pal()(length(unique(zotu$family)))), c(levels(as.factor(zotu$family))))

RANDO <- zotu %>% filter(day == 'Day9' | day =='Day18') %>%
  filter(group == 'H1_H3' | group == 'H3_H5' | group == 'H1_H5' | 
         group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL') %>%
  ggplot(., aes(x=seqid, y=relabd, fill=family)) +
  geom_bar(stat = 'identity', position = 'fill') +
  scale_fill_manual(values = bugColors_fam) + 
  labs(x = '', y = 'OTU', fill = 'log10(Relative Abundance)') +
  facet_grid(~day+group, scales = 'free') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              panel.grid = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'right',
              panel.spacing = unit(0.05, "lines"))

plotly::ggplotly(RANDO)
#ggsave("../figs/s8/s8a_HFMT_sd.pdf", width = 8.5, height = 3.5)


```

How is shannon diversity of these mice/people?
```{r}

library(vegan)
library(ggsignif)

humanized <- zotu %>% 
  filter(day == 'Day9' | species == 'human' ) %>%
  filter(relabd > .001) %>%
  select(OTU, seqid, relabd, humanized_type) %>% distinct() %>%
  reshape2::dcast(humanized_type + seqid ~ OTU, value.var = 'relabd')

humanized[is.na(humanized)] <- 0

grp_key <- humanized[,1:2] %>% 
          left_join(.,
          (zotu %>% select(seqid, species) %>% distinct()))


#create such that samples as columns, rows as features
tSNE_header <- (humanized %>% select(seqid) %>% t() %>% as.data.frame())[1,]


tSNE_input <- (humanized %>% select(-seqid) %>% t() %>% as.data.frame())[-1,] %>% 
              mutate_all(., function(x) as.numeric(as.character(x))) 

names(tSNE_input) <- tSNE_header[1,]

#now calculate shannon diversity

set.seed(123)
sd_df<- tSNE_input %>% t() %>% vegan::diversity(index = 'shannon')
     

sd_lab <- tSNE_header %>% t()

shannonD <- sd_df %>% as.data.frame() %>% cbind(., sd_lab) %>% select(shannD = '.', seqid) 

#normalize by log(k) where k is number of species over 0.05
numk <- zotu %>%# filter(day == 0, OTU != 'Otu10') %>% 
  select(OTU, seqid, relabd) %>% 
  melt(id.vars = c('seqid', 'OTU')) %>% 
  filter(value > .001) %>% 
  select(-value) %>% distinct() %>% 
  group_by(seqid) %>% mutate(kval = log2(n())) %>%
  ungroup() %>% select(seqid, kval) %>% distinct() %>% na.omit()

shannonD <- left_join(shannonD, numk) %>% mutate(shann_entropy = shannD/kval) %>% left_join(.,grp_key)

#all side by side
s9b <- ggplot(shannonD, aes(humanized_type, shann_entropy, fill = humanized_type)) +
  labs(y = 'Normalized Shannon Entropy', x = '') +
  geom_boxplot() + geom_jitter() +
  scale_fill_manual(values = c('#f05a2a', '#1b75ba', '#f9ae42')) +
  facet_wrap(~species) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 16, color = 'black'), 
        axis.text.y = element_text(size = 16, color = 'black'),
          strip.text = element_text(size = 18),
          panel.background = element_blank(), 
          strip.background = element_blank(),
           legend.position = 'none')

s9b

ggsave("../figs/s8/s8b_HFMT_sd.pdf", width = 4.5, height = 3.5)


```

PCA of D9 mice samples vs humans
```{r}
set.seed(123)

humanized <- zotu %>% 
    filter(day == 'Day9' | species == 'human') %>%
    group_by(humanized_type, OTU) %>% filter(OTU != 'Otu1') %>% 
  select(OTU, seqid, log10relabd) %>% distinct() %>% reshape2::dcast(OTU~seqid)

 humanized.matrix <- as.matrix(humanized[, -c(1)])


#actual sample labels
#tSNE_header <- names(humanized[,-1])

#simulated labels
tSNE_header <- c("H1 (Mouse)", "H1 (Mouse)", "H1 (Mouse)", 
                 "H3 (Mouse)", "H3 (Mouse)", "H3 (Mouse)",
                 "H5 (Mouse)", "H5 (Mouse)", "H5 (Mouse)", 
                 "H1 (Human)", "H3 (Human)", "H3 (Human)", "H5 (Human)")

colors <- c('#f15b2a','#ff9966','#1d76bb','#6699cc',
               '#faaf43','#ffcc66')

s9c <- M3C::pca(humanized.matrix, 
                labels = tSNE_header,
                colvec = colors,
    controlscale=TRUE,
    scale = 3, dotsize = 3,  axistextsize = 12, 
    legendtextsize = 10)


plotly::ggplotly(s9c)

ggsave("../figs/s8/s8c_HFMT_PCA.png", width = 5, height = 3.5)
ggsave("../figs/s8/s8c_HFMT_PCA.pdf", width = 5, height = 3.5)

```


Plot supplement s9
```{r}
dummy_plot <- ggplot() + theme_void()
plot_grid(s9a2, plot_grid(s9b, s9c, plot.new(), rel_widths = c(.5,1,1)), nrow = 2, rel_heights = c(2,1.5))

ggsave("../figs/s8/s8_HFMT.pdf", width = 9, height = 6)
ggsave("../figs/s8/s8_HFMT.png", width = 9, height = 6)

  
```



Now do this for heatmap with reduced  mice only (Figure 7B)
```{r}
zotuHM <- temp_zotu   #bring back og zotu table

#Only look at microves with .5% abundance in any any sample
otus.to.keep <- zotuHM %>% filter(species == 'mouse') %>%
  group_by(OTU, humanized_type) %>%
  summarise(avg_abd = max(relabd)) %>%
  filter(avg_abd > .01) %>% ungroup() %>%
  select(OTU) %>% distinct() %>%
  as_vector() 


# Drop OTUs filtered in previous step
zotuHM <- zotuHM %>%
  filter(OTU %in% otus.to.keep)

#Append taxonomic information
gtable <- tax %>%
  select(OTU, genus, family, order)

zotuHM <- zotuHM %>%
  left_join(gtable, by="OTU") %>%
  mutate(otu_genus = paste(OTU, genus, sep="_"))

```



Fig 7B - Humanized mouse FMT
```{r}

library('ggdendro')
library('grid')

set.seed(123)

humanized <- zotuHM %>% 
    #filter(day == 'Day18') %>%
     filter(day == 'Day9') %>%
    filter(group == 'H1_H3') %>%
    #filter(humanized_type == 'H1' | humanized_type == 'H3') %>%
    group_by(humanized_type, OTU) %>% filter(OTU != 'Otu1') %>% 
    summarise(mean_log10relabd = mean(log10relabd)) %>% 
  select(OTU, humanized_type, mean_log10relabd) %>% distinct() %>% reshape2::dcast(OTU~humanized_type)


humanized.scaled <- humanized
#humanized.scaled[, c(2:6)] <- (humanized.scaled[, 2:6]) #don't scale here actually

#clustering
humanized.matrix <- as.matrix(humanized.scaled[, -c(1)])
rownames(humanized.matrix) <- humanized.scaled$OTU
humanized.dendro <- as.dendrogram(hclust(d = dist(x = humanized.matrix)))

# Create dendro
dendro.plot <- ggdendrogram(data = humanized.dendro, rotate = TRUE)

# Preview the plot
print(dendro.plot)


#reorder
humanized.order <- order.dendrogram(humanized.dendro)
#humanized.long <- left_join(humanized.long, gtable)
zotuHM$OTU <- factor(x = zotuHM$OTU,
                               levels = humanized.scaled$OTU[humanized.order], 
                               ordered = TRUE)

bugColors <- setNames(c(scales::hue_pal()(length(unique(zotuHM$order)))), c(levels(as.factor(zotuHM$order))))

#zotu$tx_group <- factor(zotu$tx_group,      # Reordering group factor levels
#                         levels = c("Jax Control", "Jax Cohoused", "Envigo Control"))


#NEED TO REPEAT THIS FOR each
#fam lvl
heatmap.plot <- zotuHM %>% 
  filter(day == 'Day9' | day == 'Day18' ) %>%
  #filter(group == 'H1_H3') %>%
  ggplot(., aes(x=humanized_type, y=OTU, fill=log10relabd)) +
  geom_tile(color = 'white') +
#  scale_fill_gradientn(colours = pal) + 
  scale_fill_gradient2(high = 'firebrick3', mid = 'white', low = '#1357a6', midpoint = -3) +
  labs(x = 'Human Donor', y = 'OTU', fill = 'log10(Relative Abundance)') +
  facet_grid(~(factor(day, levels = c('Day9','Day18')))+factor(group, levels = c('H1_H3', 'H1_H5', 'H3_H5')), scales = 'free') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              axis.text.y = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'bottom',
              panel.spacing = unit(0.05, "lines"))
heatmap.plot

fam <- ggplot(data = zotuHM, aes(x = '', y = OTU)) +
    geom_tile(aes(fill = order), color = 'white') + 
  scale_fill_manual(values = bugColors) +
    labs(fill = 'Order') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_blank(), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))

cowplot::plot_grid(heatmap.plot, fam, ncol =2, rel_widths = c(.2,.3))
ggsave("../figs/fig6/6B_HFMT.png",  width = 12, height = 6)
ggsave("../figs/fig6/6B_HFMT.svg",  width = 12, height = 6)

```

6C - PCA of before/after FMT
PCA of D9 mice samples vs D18 sample
```{r}
set.seed(123)

humanized <- zotu %>% 
    filter(day == 'Day9' | day == 'Day18') %>%
    group_by(humanized_type, OTU) %>% filter(OTU != 'Otu1') %>% 
  select(OTU, seqid, log10relabd) %>% distinct() %>% reshape2::dcast(OTU~seqid)

 humanized.matrix <- as.matrix(humanized[, -c(1)])


#actual sample labels
#tSNE_header <- names(humanized[,-1])

#simulated labels
tSNE_header <- c("M1>M3 D9", "M1>M5 D9", "M3>M1 D9",
                 "M3>M5 D9", "M5>M1 D9", "M5>M3 D9",
                 "M1 D0", "M1 D0", "M1 D0", 
                 "M3 D0", "M3 D0", "M3 D0",
                 "M5 D0", "M5 D0", "M5 D0")

colors <- c('#D90000','#994FB2','#F58024','#1181EC', '#994FB2','#7BB662' , '#D7B236', '#F58024', '#7BB662')

#colors <- c('violet', 'forestgreen', 'violet', 
#            'black', 'forestgreen', 'black',
#            'firebrick','#1d76bb','#faaf43')

s9c <- M3C::pca(humanized.matrix, 
                labels = tSNE_header,
                colvec = colors,
    controlscale=TRUE,
    scale = 3, dotsize = 3,  axistextsize = 12, 
    legendtextsize = 10)


plotly::ggplotly(s9c)

ggsave("../figs/fig6/6c_postFMT_PCA.png", width = 5, height = 3.5)
ggsave("../figs/fig6/6c_postFMT_PCA.pdf", width = 5, height = 3.5)

```

Figure 6C/6D: 
```{r}

#Compare D0 vs D5 pairwise cohousing
temp <- zotu %>% filter(day == 'Day9' | day == 'Day18') %>%
                 filter(group != 'H1_CTRL' | group != 'H3_CTRL' | group != 'H5_CTRL') %>%
                  group_by(group, OTU, day, humanized_type) %>% 
            mutate(meanrelabd = mean(relabd)) %>% ungroup() %>%
            select(OTU, meanrelabd, day, group, humanized_type) %>% distinct() %>% 
            separate(group, into = c("mousex", "mousey"), sep = '_') #%>% rbind(.,)



#calculate d0 abundance
d0 <- temp %>% mutate(donor = ifelse(mousex == humanized_type, mousey, mousex),
                    recipient = humanized_type) %>% 
                    filter(day == "Day9") %>% 
                    select(donor, recipient, OTU, d0_abd=meanrelabd)

#merge with d5 analysis
d5vs0 <- temp %>% mutate(donor = ifelse(mousex == humanized_type, mousey, mousex),
                    recipient = humanized_type) %>% filter(day == "Day18") %>%
                    full_join(., d0, by = c('donor', 'recipient', 'OTU')) %>% 
                    mutate(fc_abd = meanrelabd/d0_abd,
                           fc_abd = ifelse(meanrelabd < .001, 0, fc_abd)) %>% #don't count as engrafted if < .1%
                    select(fc_abd, donor, recipient, OTU, d0_abd, meanrelabd) %>% distinct()

#Reorder for consistent plotting
d5vs0$donor <- factor(d5vs0$donor,      # Reordering group factor levels
                         levels = c("H1","H3","H5"))

d5vs0$recipient <- factor(d5vs0$recipient,      # Reordering group factor levels
                          levels = c("H1","H3","H5"))

grid <- expand_grid(donor=as.factor(c("H1","H3","H5")),recipient=as.factor(c("H1","H3","H5"))) %>% mutate(num_transfers = 0)

temp <- d5vs0 %>% filter(fc_abd > 10 & meanrelabd > .001 & d0_abd < .001) %>% left_join(., gtable) %>%
  group_by(donor, recipient) %>% mutate(num_transfers = n()) %>% 
  select(num_transfers, donor, recipient) %>% distinct() %>% ungroup()

a<- rbind(temp, anti_join(grid,temp, by = c('donor', 'recipient'))) %>%
  ggplot(., aes(y=donor, x=recipient, fill=num_transfers)) + 
    geom_tile(size=0.5, fill = 'white') +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "firebrick2") +
  #scale_fill_gradientn(colors = pal) +
  geom_text(aes(label = num_transfers), size = 6) +
  labs(x = 'Recipient', y = 'Donor', fill = '# of OTUs Transferred') +
  theme_clean() +
  theme(panel.background = element_rect(fill = "white"),
              axis.text.y = element_text(size = 14),
              axis.title.x = element_text(size = 14),
              axis.title.y = element_text(size = 14),
              axis.text.x = element_text(size = 14),
              strip.text = element_text(size = 10),
              panel.spacing = unit(0.2, "lines"),
        legend.position = 'none',
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        plot.background = element_blank(),
        strip.background = element_blank())
a
#Barplot grid
temp <- d5vs0 %>% filter(fc_abd > 10) %>% left_join(., gtable) %>%
   distinct() %>% ungroup()

b <- temp %>%
  ggplot(., aes(x = recipient, fill = family)) + 
  #geom_bar(stat = 'identity', width = .75) + 
    geom_bar(stat="count", position = 'stack', width = .5, color = 'white') +
  facet_grid(factor(donor, levels = rev(c('H1', 'H3', 'H5')))~recipient, scales = 'free_x', switch='both') +
  labs(x = 'Recipient', y = '# OTUs Transferred', fill = 'Family') +
 # scale_fill_manual(values = bugColors) +
  scale_y_continuous(breaks = c(0,10,20,30), position = 'right') +
  theme_bw() +
  theme(panel.background = element_rect(fill = "white"),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size = 14),
              axis.title.y = element_text(size = 14),
              axis.text.x = element_blank(),
              #axis.ticks = element_blank(),
              strip.text.y = element_text(angle = 90, size = 12),
              strip.text.x = element_text(size = 12),
              panel.spacing = unit(0.2, "lines"),
        legend.position = 'none',
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        strip.background = element_blank()) 
b
plotly::ggplotly(b)
plot_grid(a,b, ncol = 2)

ggsave("../figs/fig6/6C_cohouse_square.png",  width = 6, height = 3)
ggsave("../figs/fig6/6C_cohouse_square.pdf",  width = 6, height = 3)




```


```{r}
#What orders are most represented?

temp <- zotu %>% filter(day == 'Day9') %>%
                 filter(group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL') %>%
                   #filter(species == 'human') %>%
                  group_by(OTU, humanized_type) %>% 
            mutate(meanrelabd = mean(relabd)) %>% ungroup() %>% filter(meanrelabd > .001) %>%
            select(OTU, meanrelabd, humanized_type, order) %>% distinct() %>% group_by(humanized_type, order) %>% mutate(num_order = n(), abd_order = sum(meanrelabd)) %>% ungroup %>% select(-OTU, -meanrelabd) %>% distinct() 

colors <- c('#f15b2a','#1d76bb',
               '#D7B236')

a<- ggplot(temp, aes(order, num_order, fill = humanized_type), color = 'black') +
  geom_bar(stat='identity', position = 'dodge', color = 'black') + 
  theme_cowplot() + labs(x='Order', y = '# of OTUs', fill = 'Donor') +
  scale_fill_manual(values = colors) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size=12)) 

a

ggsave("../figs/s8/S8D_numSpecies.png",  width = 5, height = 3)
ggsave("../figs/s8/S8D_numSpecies.pdf",  width = 5, height = 3)

#Are differences in clostridiales significant?
h1 <-zotu %>% filter(group == 'H1_CTRL' & day == 'Day9' & order == 'Clostridiales')
h5 <- zotu %>% filter(group == 'H5_CTRL' & day == 'Day9' & order == 'Clostridiales')
h3 <- zotu %>% filter(group == 'H3_CTRL' & day == 'Day9' & order == 'Clostridiales')

wilcox.test(h1$log10relabd, h5$log10relabd, alternative = 'two.sided')
wilcox.test(h3$log10relabd, h5$log10relabd, alternative = 'two.sided')




#How many otus for H5 after cohousing
temp <- zotu %>% filter(group == 'H1_H5' | group == 'H3_H5' ) %>% 
  filter(day == 'Day18') %>%
  rbind(., zotu %>% filter(day == 'Day9' & group == 'H5_CTRL')) %>% 
  filter(humanized_type == 'H5') %>%
                  group_by(OTU, group) %>% 
            mutate(meanrelabd = mean(relabd)) %>% ungroup() %>% filter(meanrelabd > .001) %>%
            select(OTU, meanrelabd, group, order) %>% distinct() %>% 
    group_by(group, order) %>% 
  mutate(num_order = n(), abd_order = sum(meanrelabd)) %>% ungroup %>% select(-OTU, -meanrelabd) %>% distinct() 

colors <- c('#F58024','#7BB662', '#D7B236')

b<-ggplot(temp, aes(order, num_order, fill = group), color = 'black') +
  geom_bar(stat='identity', position = 'dodge', color = 'black') + 
  theme_cowplot() + labs(x='Order', y = '# of OTUs', fill = 'Donor') +
  scale_fill_manual(values = colors) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size=12)) 


plot_grid(a,b, nrow = 2)

```

Look at how bacteroides Clostridiales erysipelotrichales change between FMT
```{r}

temp <- zotu %>% filter(day == 'Day9' | day == 'Day18') %>%
  filter(species == 'mouse') %>%
  filter(order == 'Bacteroidales' | 
           order == 'Clostridiales' | 
           order == 'Erysipelotrichales') %>% 
group_by(group, OTU, day, humanized_type) %>% 
            mutate(meanrelabd = mean(relabd)) %>% ungroup() %>%
            select(OTU, meanrelabd, day, group, humanized_type, order) %>% distinct() 

temp %>% filter(humanized_type == 'H5') %>%
  filter(group != 'H5_CTRL') %>%
  filter(meanrelabd != 0) %>%
    mutate(day = factor(day, levels = c('Day9', 'Day18'))) %>%
  ggplot(., aes(day, log10(meanrelabd))) + geom_boxplot() + geom_jitter() +
  facet_grid(order~group) + theme_cowplot()

```

Lost OTUs due to cohousing

```{r}

#Compare D0 vs D5 pairwise cohousing
temp <- zotu %>% filter(day == 'Day9' | day == 'Day18') %>%
                 filter(group != 'H1_CTRL' | group != 'H3_CTRL' | group != 'H5_CTRL') %>%
                  group_by(group, OTU, day, humanized_type) %>% 
            mutate(meanrelabd = mean(relabd)) %>% ungroup() %>%
            select(OTU, meanrelabd, day, group, humanized_type) %>% distinct() %>% 
            separate(group, into = c("mousex", "mousey"), sep = '_') #%>% rbind(.,)



#calculate d0 abundance
d0 <- temp %>% mutate(donor = ifelse(mousex == humanized_type, mousey, mousex),
                    recipient = humanized_type) %>% 
                    filter(day == "Day9") %>% 
                    select(donor, recipient, OTU, d0_abd=meanrelabd)

#merge with d5 analysis
d5vs0 <- temp %>% mutate(donor = ifelse(mousex == humanized_type, mousey, mousex),
                    recipient = humanized_type) %>% filter(day == "Day18") %>%
                    full_join(., d0, by = c('donor', 'recipient', 'OTU')) %>% 
                    mutate(fc_abd = d0_abd/meanrelabd,
                           fc_abd = ifelse(d0_abd < .001, 0, fc_abd)) %>% #don't count as engrafted if < .1%
                    select(fc_abd, donor, recipient, OTU, d0_abd, meanrelabd) %>% distinct()

#Reorder for consistent plotting
d5vs0$donor <- factor(d5vs0$donor,      # Reordering group factor levels
                         levels = c("H1","H3","H5"))

d5vs0$recipient <- factor(d5vs0$recipient,      # Reordering group factor levels
                          levels = c("H1","H3","H5"))

grid <- expand_grid(donor=as.factor(c("H1","H3","H5")),recipient=as.factor(c("H1","H3","H5"))) %>% mutate(num_transfers = 0)

temp <- d5vs0 %>% filter(fc_abd > 10 & meanrelabd < .001 & d0_abd > .001 ) %>% left_join(., gtable) %>%
  group_by(donor, recipient) %>% mutate(num_transfers = n()) %>% 
  select(num_transfers, donor, recipient) %>% distinct() %>% ungroup()

a<- rbind(temp, anti_join(grid,temp, by = c('donor', 'recipient'))) %>%
  ggplot(., aes(y=donor, x=recipient, fill=num_transfers)) + 
    geom_tile(size=0.5, fill = 'white') +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "firebrick2") +
  #scale_fill_gradientn(colors = pal) +
  geom_text(aes(label = num_transfers), size = 6) +
  labs(x = 'Recipient', y = 'Donor', fill = '# of OTUs Transferred') +
  theme_clean() +
  theme(panel.background = element_rect(fill = "white"),
              axis.text.y = element_text(size = 14),
              axis.title.x = element_text(size = 14),
              axis.title.y = element_text(size = 14),
              axis.text.x = element_text(size = 14),
              strip.text = element_text(size = 10),
              panel.spacing = unit(0.2, "lines"),
        legend.position = 'none',
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        plot.background = element_blank(),
        strip.background = element_blank())
a
#Barplot grid
temp <- d5vs0 %>% filter(fc_abd > 10) %>% left_join(., gtable) %>%
   distinct() %>% ungroup()

b <- temp %>%
  ggplot(., aes(x = recipient, fill = order)) + 
  #geom_bar(stat = 'identity', width = .75) + 
    geom_bar(stat="count", position = 'stack', width = .5, color = 'white') +
  facet_grid(factor(donor, levels = rev(c('H1', 'H3', 'H5')))~recipient, scales = 'free_x', switch='both') +
  labs(x = 'Recipient', y = '# OTUs Lost', fill = 'Family') +
  scale_fill_manual(values = bugColors) +
  scale_y_continuous(breaks = c(0,10,20,30), position = 'right') +
  theme_bw() +
  theme(panel.background = element_rect(fill = "white"),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size = 14),
              axis.title.y = element_text(size = 14),
              axis.text.x = element_blank(),
              #axis.ticks = element_blank(),
              strip.text.y = element_text(angle = 90, size = 12),
              strip.text.x = element_text(size = 12),
              panel.spacing = unit(0.2, "lines"),
        legend.position = 'none',
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.background = element_blank(),
        strip.background = element_blank()) 
b
plotly::ggplotly(b)
plot_grid(a,b, ncol = 2)

ggsave("../figs/fig6/6lost_cohouse_square.png",  width = 6, height = 3)
ggsave("../figs/fig6/6lost_cohouse_square.pdf",  width = 6, height = 3)


```

Can also make supplement to compare humanized mice
```{r}


library('ggdendro')
library('grid')

set.seed(123)

humanized <- zotu %>%  filter(day == 'Day9') %>%
  filter(group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL')  %>%
    group_by(humanized_type, OTU) %>% filter(OTU != 'Otu1') %>% 
    summarise(mean_log10relabd = mean(log10relabd)) %>% 
  select(OTU, humanized_type, mean_log10relabd) %>% distinct() %>% reshape2::dcast(OTU~humanized_type)


humanized.scaled <- humanized
#humanized.scaled[, c(2:6)] <- (humanized.scaled[, 2:6]) #don't scale here actually

#clustering
humanized.matrix <- as.matrix(humanized.scaled[, -c(1)])
rownames(humanized.matrix) <- humanized.scaled$OTU
humanized.dendro <- as.dendrogram(hclust(d = dist(x = humanized.matrix)))

# Create dendro
dendro.plot <- ggdendrogram(data = humanized.dendro, rotate = TRUE)

# Preview the plot
print(dendro.plot)


#reorder
humanized.order <- order.dendrogram(humanized.dendro)
#humanized.long <- left_join(humanized.long, gtable)
zotu$OTU <- factor(x = zotu$OTU,
                               levels = humanized.scaled$OTU[humanized.order], 
                               ordered = TRUE)

bugColors <- setNames(c(scales::hue_pal()(length(unique(zotu$order)))), c(levels(as.factor(zotu$order))))


heatmap.plot <- zotu %>% filter(day == 'Day9') %>%
  filter(group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL') %>%
  ggplot(., aes(x=humanized_type, y=OTU, fill=log10relabd)) +
  geom_tile(color = 'white') +
#  scale_fill_gradientn(colours = pal) + 
  scale_fill_gradient2(high = 'firebrick3', mid = 'white', low = '#1357a6', midpoint = -3) +
  labs(x = 'Human Donor', y = 'OTU', fill = 'log10(Relative Abundance)') +
  #facet_grid(~day+group, scales = 'free') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10),
              axis.text.y = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'bottom',
              panel.spacing = unit(0.05, "lines"))

fam <- ggplot(data = zotu, aes(x = '', y = OTU)) +
    geom_tile(aes(fill = order), color = 'white') + 
    labs(fill = 'Order') +
    scale_fill_manual(values = bugColors) + #need to remove color #5 since not found in these samples
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_blank(), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))

heatmap.plot

cowplot::plot_grid(heatmap.plot, fam, ncol =2, rel_widths = c(.2,.3))
ggsave("../figs/s9/s9C_HFMT.png",  width = 12, height = 6)
ggsave("../figs/s9/s9C_HFMT.svg",  width = 12, height = 6)

```




BONUS: Show transfer of OTUs over time
Hierarchical clustering

Try ggdendro
```{r}
library('ggdendro')
library('grid')

set.seed(123)


temp <- mutate(zotu, day = as.numeric(gsub('Day', '', zotu$day)))

#storage <- c("H1_CTRL", "H3_CTRL", "H5_CTRL")
storage <- c("H1", "H3", "H5")

myplots <- vector('list', 3)
for (i in 1:3){
  
  mus_group <- storage[i]


jax.cohoused.scaled <- temp %>% 
#  filter(group == mus_group) %>%  
  filter(humanized_type == mus_group & day < 11) %>%
    group_by(day, OTU) %>% filter(OTU != 'Otu1') %>% 
  summarise(mean_log10relabd = mean(log10relabd)) %>% 
  select(OTU, day, mean_log10relabd) %>% distinct() %>% reshape2::dcast(OTU~day)
  

#clustering
jax.cohoused.matrix <- as.matrix(jax.cohoused.scaled[, -c(1)])
rownames(jax.cohoused.matrix) <- jax.cohoused.scaled$OTU
jax.cohoused.dendro <- as.dendrogram(hclust(d = dist(x = jax.cohoused.matrix)))


##Heatmap
jax.cohoused.long <- reshape2::melt(jax.cohoused.scaled, id = c("OTU"))

#reorder
jax.cohoused.order <- order.dendrogram(jax.cohoused.dendro)
jax.cohoused.long <- left_join(jax.cohoused.long, gtable)
jax.cohoused.long$OTU <- factor(x = jax.cohoused.long$OTU,
                               levels = jax.cohoused.scaled$OTU[jax.cohoused.order], 
                               ordered = TRUE)

bugColors <- setNames(c(scales::hue_pal()(length(unique(jax.cohoused.long$family)))), c(levels(as.factor(jax.cohoused.long$family))))


# Heatmap
heatmap.plot <- ggplot(data = jax.cohoused.long, aes(x = variable, y = OTU)) +
  geom_tile(aes(fill = value)) +
  #scale_fill_gradientn(colours = pal) + 
  scale_fill_gradient2(high = 'firebrick3', mid = 'white', low = '#1357a6', midpoint = -3) +
  labs(x='Day', fill ='Relative Abundance', title = mus_group) +
  theme(strip.text.y.right = element_text(angle = 0),
        panel.background = element_rect(fill = "white"), 
              axis.ticks = element_blank(), 
              axis.title.y = element_blank(),
              axis.text.y = element_blank(), 
              axis.text.x = element_text(size = 10),
              strip.text = element_text(size = 10),
              panel.spacing = unit(0.05, "lines"),
        legend.position = 'left') 

fam <- ggplot(data = jax.cohoused.long, aes(x = '', y = OTU)) + 
    geom_tile(aes(fill = family)) +
    labs(fill = 'Family') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_rect(fill = "white"), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))

myplots[[i]] <- cowplot::plot_grid(heatmap.plot, fam, 
                                   ncol =2, rel_widths = c(.5,1))

#plot_list[i] <- heatmap.plot
}

plot_grid(myplots[[1]], myplots[[2]], myplots[[3]], ncol = 1)

ggsave("../figs/s9/hFMT_initialColonization.pdf",  width = 12, height = 12)


```



Look at trajectory of Ctrl mice
```{r}

library('ggdendro')
library('grid')

set.seed(123)

humanized <- zotu %>% 
  filter(group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL') %>% na.omit() %>%
#     filter(day == 'Day6' | species == 'human') %>%
    filter(species == 'mouse') %>%
    group_by(humanized_type, OTU) %>% filter(OTU != 'Otu1') %>% 
    summarise(mean_log10relabd = mean(log10relabd)) %>% 
  select(OTU, humanized_type, mean_log10relabd) %>% distinct() %>% reshape2::dcast(OTU~humanized_type)


humanized.scaled <- humanized
#humanized.scaled[, c(2:6)] <- (humanized.scaled[, 2:6]) #don't scale here actually

#clustering
humanized.matrix <- as.matrix(humanized.scaled[, -c(1)])
rownames(humanized.matrix) <- humanized.scaled$OTU
humanized.dendro <- as.dendrogram(hclust(d = dist(x = humanized.matrix)))

# Create dendro
dendro.plot <- ggdendrogram(data = humanized.dendro, rotate = TRUE)

# Preview the plot
print(dendro.plot)


#reorder
humanized.order <- order.dendrogram(humanized.dendro)
#humanized.long <- left_join(humanized.long, gtable)
zotu$OTU <- factor(x = zotu$OTU,
                               levels = humanized.scaled$OTU[humanized.order], 
                               ordered = TRUE)

bugColors <- setNames(c(scales::hue_pal()(length(unique(zotu$family)))), c(levels(as.factor(zotu$family))))

#zotu$tx_group <- factor(zotu$tx_group,      # Reordering group factor levels
#                         levels = c("Jax Control", "Jax Cohoused", "Envigo Control"))

#fam lvl
heatmap.plot <- zotu %>% na.omit() %>%
  filter(group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL' | species == 'human') %>%
 # group_by(humanized_type, family, OTU, day) %>%
  #mutate(mean_log10relabd = mean(log10relabd)) %>% ungroup() %>%
  ggplot(., aes(x=day, y=OTU, fill=log10relabd)) +
  geom_tile() +
  scale_fill_gradientn(colours = pal) + 
  labs(x = '', y = 'OTU', fill = 'log10(Relative Abundance)') +
  facet_grid(~humanized_type, scales = 'free') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              axis.text.y = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'bottom',
              panel.spacing = unit(0.05, "lines"))

heatmap.plot

fam <- ggplot(data = zotu, aes(x = '', y = OTU)) +
    geom_tile(aes(fill = order)) + 
    labs(fill = 'Family') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_blank(), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(), 
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))


cowplot::plot_grid(heatmap.plot, fam, ncol =2, rel_widths = c(.2,.3))
#ggsave("../figs/s9/s9A_HFMT.png",  width = 12, height = 6)
#ggsave("../figs/s9/s9A_HFMT.svg",  width = 12, height = 6)

```

Compare D9 mice to human samples (heatmap)
```{}

library('ggdendro')
library('grid')

set.seed(123)

humanized <- zotu %>% 
     filter(species == 'human') %>%
    group_by(humanized_type, OTU) %>%
    summarise(mean_log10relabd = mean(log10relabd)) %>% 
  select(OTU, humanized_type, mean_log10relabd) %>% distinct() %>% reshape2::dcast(OTU~humanized_type)


humanized.scaled <- humanized
#humanized.scaled[, c(2:6)] <- (humanized.scaled[, 2:6]) #don't scale here actually

#clustering
humanized.matrix <- as.matrix(humanized.scaled[, -c(1)])
rownames(humanized.matrix) <- humanized.scaled$OTU
humanized.dendro <- as.dendrogram(hclust(d = dist(x = humanized.matrix)))

# Create dendro
dendro.plot <- ggdendrogram(data = humanized.dendro, rotate = TRUE)

# Preview the plot
print(dendro.plot)


#reorder by cluster
humanized.order <- order.dendrogram(humanized.dendro)

#reorder by phylogeny
#zotu.order <- zotu %>% select(OTU, order) %>% arrange(order) %>% select(OTU) %>% distinct
#zotu$OTU <- factor(zotu$OTU, levels = zotu.order$OTU)

#zotu$OTU <- factor(zotu$OTU, levels = humanized.order$OTU)

bugColors_fam <- setNames(c(scales::hue_pal()(length(unique(zotu$family)))), c(levels(as.factor(zotu$family))))


#fam lvl
heatmap.plot <- zotu %>% filter(day == 'Day9' | species == 'human' ) %>%
  filter(group == 'H1_CTRL' | group == 'H3_CTRL' | group == 'H5_CTRL' | species == 'human') %>%
 # group_by(humanized_type, family, OTU, day) %>%
  #mutate(mean_log10relabd = mean(log10relabd)) %>% ungroup() %>%
  ggplot(., aes(x=species, y=OTU, fill=log10relabd)) +
  geom_tile() +
  scale_fill_gradientn(colours = pal) + 
  labs(x = '', y = 'OTU', fill = 'log10(Relative Abundance)') +
  facet_grid(~humanized_type, scales = 'free') +
  theme(strip.text.y.right = element_text(angle = 0, size = 10),
        strip.text.x.top = element_text(angle = 0, size = 10),
              axis.ticks = element_blank(), 
              axis.text.x = element_text(size = 10, hjust = 1, angle = 45),
              axis.text.y = element_blank(),
              panel.grid = element_blank(),
              strip.text = element_text(size = 6),
              panel.background = element_blank(), 
              strip.background = element_blank(),
              legend.position = 'bottom',
              panel.spacing = unit(0.05, "lines"))

fam <- zotu %>% ggplot(., aes(x = '', y=OTU)) +
    geom_tile(aes(fill = order)) + 
    labs(fill = 'Order') +
    theme(strip.text.y.right = element_text(angle = 0),
              panel.background = element_blank(), 
              axis.ticks = element_blank(), 
              axis.text.y = element_blank(),
              axis.title.y = element_blank(),
              axis.text.x = element_blank(),
              panel.grid = element_blank(),
              axis.title.x = element_blank(),
              strip.text = element_blank(),
              panel.spacing = unit(0.05, "lines"))


s9a <- cowplot::plot_grid(heatmap.plot, fam, ncol=2, rel_widths = c(.2,.3), rel_heights = c(1,.4))

ggsave("../figs/s9/s9A_HFMT.png",  width = 12, height = 6)
ggsave("../figs/s9/s9A_HFMT.svg",  width = 12, height = 6)

```