---
title: "Fig4_Metagenomics"
output: html_document
---
```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)

setwd("~/Dropbox/fmt/scripts/")
library(tidyverse)
library(wesanderson)
library(ggthemes)
library(reshape2)

ven <- c('#FAAF40',"#1B75BB","#37B34A", "#F05A28")
pal <- wes_palette("Zissou1", 100, type = "continuous")


options(stringsAsFactors = F)
options(scipen = 10000)

```



Let's look at Metagenomic sequencing across the gut compartment?
```{r}

#Load in RPKM mapped to KO terms
noFMT_ko <- read.csv("../experiments/2021-07-19_Lumen_Metagenomics/data/rpkm_all_samples_summedbyko.csv")
names(noFMT_ko) <- c("ko_map", "noFMT_1_duodenum",'noFMT_1_jejunum', 'noFMT_1_ileum','noFMT_1_cecum', 'noFMT_1_colon',
                    'noFMT_2_duodenum','noFMT_2_jejunum', 'noFMT_2_ileum','noFMT_2_cecum', 'noFMT_2_colon',
                    'noFMT_3_duodenum','noFMT_3_jejunum', 'noFMT_3_ileum','noFMT_3_cecum', 'noFMT_3_colon',
                    'noFMT_4_duodenum','noFMT_4_jejunum', 'noFMT_4_ileum','noFMT_4_cecum', 'noFMT_4_colon',
                    "donor_1_duodenum",'donor_1_jejunum', 'donor_1_ileum','donor_1_cecum', 'donor_1_colon',
                    'donor_2_duodenum','donor_2_jejunum', 'donor_2_ileum','donor_2_cecum', 'donor_2_colon',
                    'donor_3_duodenum','donor_3_jejunum', 'donor_3_ileum','donor_3_cecum', 'donor_3_colon',
                    'donor_4_duodenum','donor_4_jejunum', 'donor_4_ileum','donor_4_cecum', 'donor_4_colon')

postFMT_ko <- read.csv("../experiments/2021-07-19_Lumen_Metagenomics/data/rpkm_all_samples_summedbyko_Post.csv")
names(postFMT_ko) <- c("ko_map", "postFMT_1_duodenum",'postFMT_1_jejunum', 'postFMT_1_ileum','postFMT_1_cecum', 'postFMT_1_colon',
                    'postFMT_2_duodenum','postFMT_2_jejunum', 'postFMT_2_ileum','postFMT_2_cecum', 'postFMT_2_colon',
                    'postFMT_3_duodenum','postFMT_3_jejunum', 'postFMT_3_ileum','postFMT_3_cecum', 'postFMT_3_colon',
                    'postFMT_4_duodenum','postFMT_4_jejunum', 'postFMT_4_ileum','postFMT_4_cecum', 'postFMT_4_colon')


#merge datasets
KO_counts <- noFMT_ko %>% full_join(., postFMT_ko, by = 'ko_map') %>% 
  filter(ko_map !='Unclassified') %>% replace(is.na(.), 0) %>%
  mutate(rowsum = rowSums(.[2:61])) %>% filter(rowsum > 1000)

#subset to only most prevalent 

KO_counts[KO_counts == 0] <- 1 #add pseudocounts

```


Cleaned up data frame for Thomas
```{r}

#Load in RPKM mapped to KO terms
noFMT_counts <- read.csv("../experiments/2021-07-19_Lumen_Metagenomics/data/read_counts_all_samples_summedbyko_pre.csv")
names(noFMT_counts) <- c("ko_map", "noFMT_1_duodenum",'noFMT_1_jejunum', 'noFMT_1_ileum','noFMT_1_cecum', 'noFMT_1_colon',
                    'noFMT_2_duodenum','noFMT_2_jejunum', 'noFMT_2_ileum','noFMT_2_cecum', 'noFMT_2_colon',
                    'noFMT_3_duodenum','noFMT_3_jejunum', 'noFMT_3_ileum','noFMT_3_cecum', 'noFMT_3_colon',
                    'noFMT_4_duodenum','noFMT_4_jejunum', 'noFMT_4_ileum','noFMT_4_cecum', 'noFMT_4_colon',
                    "donor_1_duodenum",'donor_1_jejunum', 'donor_1_ileum','donor_1_cecum', 'donor_1_colon',
                    'donor_2_duodenum','donor_2_jejunum', 'donor_2_ileum','donor_2_cecum', 'donor_2_colon',
                    'donor_3_duodenum','donor_3_jejunum', 'donor_3_ileum','donor_3_cecum', 'donor_3_colon',
                    'donor_4_duodenum','donor_4_jejunum', 'donor_4_ileum','donor_4_cecum', 'donor_4_colon')

postFMT_counts <- read.csv("../experiments/2021-07-19_Lumen_Metagenomics/data/read_counts_all_samples_summedbyko_post.csv")
names(postFMT_counts) <- c("ko_map", 
                    "postFMT_1_duodenum",'postFMT_1_jejunum', 'postFMT_1_ileum','postFMT_1_cecum', 'postFMT_1_colon',
                    'postFMT_2_duodenum','postFMT_2_jejunum', 'postFMT_2_ileum','postFMT_2_cecum', 'postFMT_2_colon',
                    'postFMT_3_duodenum','postFMT_3_jejunum', 'postFMT_3_ileum','postFMT_3_cecum', 'postFMT_3_colon',
                    'postFMT_4_duodenum','postFMT_4_jejunum', 'postFMT_4_ileum','postFMT_4_cecum', 'postFMT_4_colon')


kos <- rbind(select(noFMT_counts, ko_map), select(postFMT_counts, ko_map)) %>% distinct()

# left_join(kos, noFMT_counts) %>% left_join(., postFMT_counts) %>% #replace(is.na(.), 1) %>%
#      write.csv("../experiments/2021-07-19_Lumen_Metagenomics/data/fmt_GI_spatialMetagenomics_Genecounts.csv")


```


Let's try clustering (Doesn't really work)
```{r}

tSNE_input <- (KO_counts %>% select(-ko_map) %>% t() %>% as.data.frame())[-1,] %>% 
              mutate_all(., function(x) as.numeric(as.character(x))) 


s3a <- M3C::pca(tSNE_input, 
                labels = as.factor(names(KO_counts)),
    controlscale = T, scale = 3, dotsize = 3,  axistextsize = 12, 
    legendtextsize = 10)

s3a

```



Microbiome KO counts appear mostly correlated, with distinct outliers
```{r}


KO_long <- KO_counts %>% melt(id.var = 'ko_map') %>% 
          group_by(variable) %>% mutate(value = signif((value/sum(value)*10000), digits=5)) %>% ungroup() %>%#This calculates relative abundance in sample
          separate(variable, into = c('group', 'mouse_num', 'gi_location'), sep = '_', remove = T) %>% 
          group_by(group, gi_location, ko_map) %>% mutate(ave_rel_abd = mean(value)) %>% ungroup() %>% #average across mice
          select(-mouse_num, -value) %>% distinct 

noFMT <- KO_long %>% filter(group == 'noFMT') %>% rename("no_FMT_control" ="ave_rel_abd") %>% select(-group)
postFMT <- KO_long %>% filter(group == 'postFMT') %>% rename('Recipient'='ave_rel_abd') %>% select(-group)
donor <- KO_long %>% filter(group == 'donor') %>% rename('Donor'='ave_rel_abd') %>% select(-group)


full_join(noFMT, postFMT, by = c('gi_location', 'ko_map')) %>%
  full_join(., donor,  by = c('gi_location', 'ko_map')) %>% 
  select(-ko_map, -gi_location) %>%
    mutate(Donor = log2(Donor),
      Recipient = log2(Recipient),
      no_FMT_control = log2(no_FMT_control)) %>% 
      GGally::ggpairs(., columns = ) + theme_bw()

ggsave("../figs/s4/SX_Metagenomic_read_coverage.png", width = 12, height = 12)

```




Now let's look at ratios of KO terms
```{r}

KO_ratios <- full_join(noFMT, postFMT, by = c('gi_location', 'ko_map')) %>% 
              full_join(., donor,  by = c('gi_location', 'ko_map')) %>%
              mutate(
                postFMTvsnoFMT = log2(Recipient/no_FMT_control),
                donorVnoFMT = log2(Donor/no_FMT_control),
                postFMTvdonor = log2(Recipient/Donor)
              )


KO_ratios %>% ggplot(., aes(log2(no_FMT_control), y = log2(Donor))) + geom_point() +
  facet_wrap(~factor(gi_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), nrow = 1) +
  geom_abline(slope=1, linetype = 'dashed', color = 'red') +
  theme_bw() +
  labs(x='No FMT RPKMs', y = 'Donor RPKMS') +
  theme(plot.background = element_blank(),
          strip.background = element_blank(),
          axis.title.y = element_text(size = 14),
          axis.title.x = element_text(size = 14),
              axis.text.y = element_text(size = 12),
              axis.text.x = element_text(size = 12), 
              strip.text = element_text(size = 12),
              panel.spacing = unit(0.05, "lines"),
          legend.position = 'bottom')
  

```


What are most abundant & enriched genes in postFMT
```{r}

plot <- KO_ratios %>% ggplot(., aes(log2(Recipient), y = postFMTvsnoFMT, fill = ko_map)) + geom_point() +
  facet_wrap(~factor(gi_location, levels=c("duodenum","jejunum", "ileum","cecum", 'colon')), nrow = 1) +
  geom_abline(slope=1, linetype = 'dashed', color = 'red') +
  theme_bw() +
  labs(x='post FMT RPKMs', y = 'log2(Enrichment score)') +
  theme(plot.background = element_blank(),
          strip.background = element_blank(),
          axis.title.y = element_text(size = 14),
          axis.title.x = element_text(size = 14),
              axis.text.y = element_text(size = 12),
              axis.text.x = element_text(size = 12), 
              strip.text = element_text(size = 12),

                      panel.spacing = unit(0.05, "lines"),
          legend.position = 'none')

#plotly::ggplotly(plot)
plot

#Identify most Abundant KO terms in PostFMT vs no FMT
KO_ratios %>% filter(2^postFMTvsnoFMT > 10) %>%
          write.table("../experiments/2021-07-19_Lumen_Metagenomics/data/EnrichedKOs_PostVnoFMT.csv", 
          quote = F, row.names = F, sep = ',')

#Identify most Abundant KO terms in Donor vs no FMT
KO_ratios %>% filter(2^donorVnoFMT > 10) %>%
          write.table("../experiments/2021-07-19_Lumen_Metagenomics/data/EnrichedKOs_donorVnoFMT.csv", 
          quote = F, row.names = F, sep = ',')


```


Now let's try stats...
```{r}

KO_long_split <- KO_counts %>% melt(id.var = 'ko_map') %>% 
          group_by(variable) %>% 
          mutate(value = log2(signif((value/sum(value)*100000), digits=6))) %>% 
          ungroup() %>% #This calculates relative abundance in sample
          separate(variable, into = c('group', 'mouse_num', 'gi_location'), sep = '_', remove = T)

ko_list <- KO_counts$ko_map  #list of ko's to loop through

ko_stats <- as.data.frame(ko_list) %>% mutate(duodenum_p.val=0,
                                              jejunum_p.val=0,
                                              ileum_p.val=0,
                                              cecum_p.val=0,
                                              colon_p.val=0) #make data frame to store values


#duodenum  
for (i in 1:length(ko_list)) {

  ko <- ko_list[[i]]
  gi <- "duodenum"
  print(ko)

ko_stats$duodenum_p.val[[i]] <- try(t.test(filter(KO_long_split, group == 'noFMT' & gi_location == gi & ko_map == ko)$value, 
                               filter(KO_long_split, group == 'postFMT' & gi_location == gi & ko_map == ko)$value, 
                              alternative = "less")$p.val)  #one-sided t.test
  
}

#jejunum  
for (i in 1:length(ko_list)) {

  ko <- ko_list[[i]]
  gi <- "jejunum"
  print(ko)

ko_stats$jejunum_p.val[[i]] <- try(t.test(filter(KO_long_split, group == 'noFMT' & gi_location == gi & ko_map == ko)$value, 
                               filter(KO_long_split, group == 'postFMT' & gi_location == gi & ko_map == ko)$value, 
                              alternative = "less")$p.val)  #one-sided t.test
  
}
#ileum  
for (i in 1:length(ko_list)) {

  ko <- ko_list[[i]]
  gi <- "ileum"
  print(ko)

ko_stats$ileum_p.val[[i]] <- try(t.test(filter(KO_long_split, group == 'noFMT' & gi_location == gi & ko_map == ko)$value, 
                               filter(KO_long_split, group == 'postFMT' & gi_location == gi & ko_map == ko)$value, 
                              alternative = "less")$p.val)  #one-sided t.test
  
}
#cecum  
for (i in 1:length(ko_list)) {

  ko <- ko_list[[i]]
  gi <- "cecum"
  print(ko)

ko_stats$cecum_p.val[[i]] <- try(t.test(filter(KO_long_split, group == 'noFMT' & gi_location == gi & ko_map == ko)$value, 
                               filter(KO_long_split, group == 'postFMT' & gi_location == gi & ko_map == ko)$value, 
                              alternative = "less")$p.val)  #one-sided t.test
  
}


#colon  
for (i in 1:length(ko_list)) {

  ko <- ko_list[[i]]
  gi <- "colon"
  print(ko)

ko_stats$colon_p.val[[i]] <- try(t.test(filter(KO_long_split, group == 'noFMT' & gi_location == gi & ko_map == ko)$value, 
                               filter(KO_long_split, group == 'postFMT' & gi_location == gi & ko_map == ko)$value, 
                              alternative = "less")$p.val)  #one-sided t.test
  
}




```



